# Understanding our spatial data

![](images/workflows/01_counts.png){width=100%}

## Goals  

- Load the spatial data into R as a SeuratObject  
- Access and view our gene expression counts and metadata 

## Overview

```{r include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE
)
```

```{r}
library(Seurat)
library(SeuratObject)
library(tidyverse)
library(here)
```

Conducting processing and data analysis steps with open source, third-party
software/packages provides flexibility and supports a range of different
analyses. However, there are key steps undertaken before you can load your data
into R/Python.

## The raw data

![](images/30_data/data_generation.png){width=100%}

The files generated by high-resolution spatial transcriptomics platforms, such 
as the CosMx Spatial Molecular Imager (SMI), are typically categorised based on
their processing stage. The **Raw Files** contain the primary output from the
instrument's run, necessary for decoding and spatial assignment, while the
**Flat Files** represent the aggregated, cell-level data ready for statistical
analysis in R or other environments.

### Raw files

The raw files represent the initial, large-volume data output directly from the
imaging instrument after the spatial transcriptomics experiment is complete.
This directory contains all the key imaging and log information required to
computationally process the raw fluorescent signals into **quantifiable transcript
locations and cellular boundaries**.

Key raw folders and files contain essential data used for **segmentation and
cell coordinate mapping**. For example, the `CellStatsDir/Morphology2D` folder
holds the images, including those generated from immunofluorescence staining
(e.g., DAPI, PanCK, CD45), which are vital inputs for cell segmentation, as the
**raw gene expression measurements are initially at the transcript level, not the 
cell level**.

Additionally, the `RunSummary/` directory contains the instrument configuration
files necessary for accurately converting image pixels into physical slide X-Y
coordinates.
  
| Folder / File Name              | Description                                                |
|---------------------------------|------------------------------------------------------------|
| `Flowcell_Folder/`              | Root directory containing logs and slide/run data          |
| `Logs/`                         | FOV coordinate files                                       |
| `GSM7473682_HC_a_S1/`           | Main data directory for slide `GSM7473682_HC_a`, slot 1    |
| `plex_001.csv`                  | Target probes metadata                                     |
| `CellStatsDir/`                 | Segmentation outputs (labels, overlays, morphology images) |
| `CellStatsDir/`                 | Location of high-resolution images                         |
| `RunSummary/`                   | Instrument configuration files (e.g. pixel–nm ratio)       |
| `AnalysisResults/`              | Processed analysis outputs (FOV summaries, detection limits, coordinates) |

Data retrieved from 
[CosMxDACustomModules](https://github.com/Nanostring-Biostats/CosMxDACustomModules/blob/main/Export/CosMxDAExportSetup.docx). 

### Flat files

:::{.rmdimportant}

**You do not need to access every one of these files!**

Most of what we need to do processing. Typically a directory of flat files is
read in as a whole, and we don't need to worry too much. But note there are no
images in here.

:::

The flat files contain the organised, post-processing outputs from a pipeline
(like AtoMx SIP) that are directly ready for loading into R packages such as
`Seurat`. 

These files come after cell segmentation, where the **single-molecule data are
summarised into count matrices at the single-cell level**. The main file, 
`SLIDE_exprMat_file.csv.gz`, contains the counts of genes per cell.

Additionally, key metadata files are read in alongside the gene counts which
inform the spatial coordinates of FoVs (`SLIDE_fov_positions_file.csv.gz`), 
transcripts (`SLIDE_tx_file.csv.gz`), and segmented cell boundaries
(`SLIDE-polygons.csv.gz`).

| File name                         | Description                               |
|-----------------------------------|-------------------------------------------|
| `SLIDE-polygons.csv.gz`           | Cell borders                              |
| `SLIDE_exprMat_file.csv.gz`       | Counts of genes per cell (counts matrix)  |
| `SLIDE_fov_positions_file.csv.gz` | Location of FOVs on slide                 |
| `SLIDE_metadata_file.csv.gz`      | Cell-level QC metadata                    |
| `SLIDE_tx_file.csv.gz`            | Location of individual transcripts        |

:::{.rmdimportant}

**Xenium data**

For a Xenium slide there is just the one typical output directory; they discuss
the formats in the section on
[data archiving](https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-archive-data),
but again, software tools take the directory as a whole.

:::

## Importing the flat files into R

This section was conducted prior to the workshop. The directory of flat files
were loaded into R using a modified version of `Seurat::LoadNanostring()`
function (see below). For the complete code to load the data, see the [following
page](https://swbioinf.github.io/spatialsnippets/d_cosmxIBD.html#Data_load).

:::{.rmdwarning}

**A note on `Seurat::LoadNanostring()`**

To load the flat files into R, we can use the `Seurat::LoadNanostring()` function.
However, this default method **drops most of the metadata in the Seurat object**.
For example, what field of view (FoV) each cell a member of is missing. 

We provide an alternate function. In time, this should be fixed within Seurat.
See comments [here](https://github.com/satijalab/seurat/discussions/9261). This
is an example code snippet using the replacement function:

```
# Import the custom LoadNanostring() function
source(here("scripts/LoadNanostring_edited_function.R"))

# Define the path to the flat files. Note that the input is the whole directory.
# The function deals with reading and assigning the metadata correctly to the
# cells and gene counts for a single sample.
example_path <- here("raw_data", "GSM7473682_HC_a/")
example_so <- LoadNanostring(example_path, assay='RNA', fov="GSM7473682.HC.a")
```

:::

## Loading the subsampled workshop data

We use the subsampled data for this workshop, for three healthy control 
samples (HC) and three samples with Crohn's disease (CD). The first four field
of views for each sample are used.

![](images/30_data/fov_subsample.png){width=80%}

```{r}
# Load the prepared data
so_path <- here("data", "GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS")
so <- readRDS(so_path)
```

:::{.rmdimportant}

**Analysing large data sets with Sketch-based integration**

When analysing your own data, it is often necessary to work with the entire
data set rather than an arbitrary subsample, as simple subsampling can omit
important biological information or spatial regions. However, processing very
large data sets can be computationally demanding.

![](images/30_data/geosketch.png){width=100%}

Sketch-based analysis offers a practical solution by selecting a representative
subset of cells that preserves the overall structure and diversity of the full
data set. Instead of arbitrarily subsetting by fields of view (FoVs), this
approach uses geometric sketching to capture both common and rare cell
populations. You can **perform initial preprocessing steps on this smaller subset
and then extend the results back to the complete dataset for downstream analyses.**

For more details, see Seurat’s sketch-based analysis vignette[^1]$^,$[^3] and the
GeoSketch method description[^2] for a Python alternative.

[^1]: https://satijalab.org/seurat/articles/seurat5_sketch_analysis
[^2]: https://cb.csail.mit.edu/geosketch/
[^3]: https://satijalab.org/seurat/articles/parsebio_sketch_integration

:::

## Visualising our spatial data

Now that our data is loaded, we can display the spatial location of each cell.

```{r}
hc_idents <- unique(so$orig.ident)[1:3]
cd_idents <- unique(so$orig.ident)[4:6]
```

```{r}
for (i in hc_idents) {
  p <- ImageDimPlot(so, fov = i, group.by = "fov_name", axes = T)
  print(p)
}
```

```{r}
for (i in cd_idents) {
  p <- ImageDimPlot(so, fov = i, group.by = "fov_name", axes = T)
  print(p)
}
```

:::{.rmdimportant}

**Xenium does not have FOVs**

Xenium uses a wide-field fluorescence microscope for automated image
acquisition across a larger, continuous imaging area. This approach allows
Xenium to map gene expression over the entire tissue region selected by the
user, rather than stitching together multiple small, distinct FOV images.

:::

### Understanding the `SeuratObject`

:::{.rmdnote}

**Why Seurat?**

Seurat objects are a data structure to store the count data and additional metadata.
As the data is very high dimensional, it is important
that connected information about cells, samples, and data stay connected
during data analysis.

Other frameworks such as `scverse`, and data classes like `anndata` and 
`SpatialFeatureExperiment` represent the data differently, but all meet the
needs above. Differences in compatibility with other tools.

Flexibility across single-cell transcriptomics.

:::

Display the Seurat object.

```{r}
so
```

**999 features across 68624 samples with 2 assays**

- 980 genes + 19 negprobes
- 68.6k number of segmented cells analysed
- Two different types of measurements, detailed below

**Active assay: RNA (980 features, 0 variable features):**

- RNA assay: Primary gene expression measurement data (counts)
- 980 transcripts measured
- No variable features as feature selection has not been conducted yet

**6 layers present: counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, 
counts.GSM7473684.HC.c, counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, 
counts.GSM7473690.CD.c**

- Each layer represents a sample from either HC or CD donors. Required for
comparison of both groups downstream

**1 other assay present: negprobes**

This secondary assay specifically stores the counts corresponding to negative
control probes. In spatial transcriptomics, tracking these control probes is
crucial for quality control, specificity assessment, and defining background
signal.

**6 spatial fields of view present: GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b GSM7473690.CD.c:**

The list of six spatial fields of view (FOVs) explicitly links the spatial/image
information stored in the Seurat object to the six specific samples/layers
identified above. This demonstrates that the data set is ready for spatially
resolved analysis and visualisation, allowing results (like cell clusters or
gene expression) to be mapped back onto the original tissue sections for all
six integrated samples.

:::{.rmdimportant}

**CosMx FOVs and sample names**

Note that the field of view names are delimited by `.`. At the time of writing,
underscores will cause some functions to fail silently. For more information,
see [this issue](https://github.com/satijalab/seurat/issues/10103).

:::

### Exploring the metadata

Along with the transcript counts per-cell, metadata tracks other central
information such as its experimental details. This is important so we know which
groups to compare downstream (HC vs. UC). This study has one sample per slide,
but there are typically more. Examples of metadata include:

- The spatial coordinates of your cells and genes
- Segmented cell boundaries
- Which tissue sample and FOV data belongs to

```{r}
head(so@meta.data)
```

Display the metadata columns:

```{r}
names(so@meta.data)
```

There is a lot of information. Use something like `skimr` for EDA and to
understand each field at a glance.

```{r}
skimr::skim(so@meta.data)
```

The number of rows (68624) represents a cell, and the information about that
cell.

:::{.rmdimportant}

**CosMx FOV =/= SeuratObject FOV**

Note that the FOV in `SeuratObject` represents a single tissue sample, in this
case, from one donor. Recall in the data generation step that users must select
FOVs from the imaging preview. These FOVs are represented in the `SeuratObject`
as the `fov` and `fov_name` metadata.

:::

View the `tissue_sample` metadata to prepare for the next step. The following
shows the table counts the number of cells for each sample.

```{r}
table(so$tissue_sample)
```

## Count data (`assays`)

We need a way to store the results from the probe hybridisation for each sample.
The count data is represented as `assays` in a `SeuratObject`. These include the
total RNA counts and the total negative control probe counts per cell.

![](images/30_data/count_assays.png){width=100%}

```{r}
so@assays
```

### Gene expression count matrices

```{r}
so@assays$RNA$counts.GSM7473682.HC.a[1:5,1:5]
```

Each row represents a gene, and the column shows the cell for that sample.

### Negative probes

Negative control probes, or `negprobes` are non-binding probes that indicate how
much noise there is in your data. In this data, 19 `negprobes` are used. We will
explore and utilise this more in the following quality control section.

```{r}
so@assays$negprobes
```

```{r}
rownames(so@assays$negprobes)
```

```{r}
so@assays$negprobes[1:5, 1:5]
```
