<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 10 Batch Correction | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="10.1 Do we need batch correction? There are regions in the UMAP where we can see some separation by sample. This is common in spatial data, especially when working with human data. (In fact, this...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 10 Batch Correction | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="10.1 Do we need batch correction? There are regions in the UMAP where we can see some separation by sample. This is common in spatial data, especially when working with human data. (In fact, this...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 10 Batch Correction | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="10.1 Do we need batch correction? There are regions in the UMAP where we can see some separation by sample. This is common in spatial data, especially when working with human data. (In fact, this...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="exampleexperiment.html"><span class="header-section-number">4</span> Example Experiment</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="" href="input-data.html"><span class="header-section-number">6</span> Input Data</a></li>
<li><a class="" href="qc.html"><span class="header-section-number">7</span> QC</a></li>
<li><a class="" href="basic-visualisation.html"><span class="header-section-number">8</span> Basic visualisation</a></li>
<li><a class="" href="reduce-dimensionality---umaps.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="active" href="batch-correction.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="clustering-labeling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="analysis-example.html"><span class="header-section-number">13</span> Analysis example</a></li>
<li><a class="" href="analysis-example-1.html"><span class="header-section-number">14</span> Analysis example</a></li>
<li><a class="" href="next-steps.html"><span class="header-section-number">15</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">16</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="batch-correction" class="section level1" number="10">
<h1>
<span class="header-section-number">10</span> Batch Correction<a class="anchor" aria-label="anchor" href="#batch-correction"><i class="fas fa-link"></i></a>
</h1>
<div id="do-we-need-batch-correction" class="section level2" number="10.1">
<h2>
<span class="header-section-number">10.1</span> Do we need batch correction?<a class="anchor" aria-label="anchor" href="#do-we-need-batch-correction"><i class="fas fa-link"></i></a>
</h2>
<p>There are regions in the UMAP where we can see some separation by sample. This is common in spatial data, especially when working with human data.</p>
<p>(In fact, this example isn’t too bad, but for the purposes of this work, we will go through the process of batch correction).</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, group.by<span class="op">=</span><span class="st">'tissue_sample'</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-25-1.png" width="672"></div>
<p>This becomes more obvious if we split the plot by sample as well;</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, group.by<span class="op">=</span><span class="st">'tissue_sample'</span>, split.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-26-1.png" width="672"></div>
<p>If were were to continue on to cell clustering at analysis at this point, we might find that we build clusters around individual samples. We instead want to make cell clusters that form around cell ‘types’ (or transcriptionallly similar cell states) across multiple samples, which can be matched and compared across experimental groups.</p>
</div>
<div id="running-batch-correction" class="section level2" number="10.2">
<h2>
<span class="header-section-number">10.2</span> Running batch correction<a class="anchor" aria-label="anchor" href="#running-batch-correction"><i class="fas fa-link"></i></a>
</h2>
<p>We can apply a <em>batch correction</em> or <em>integration</em> method to adjust our principal components to deemphasise the betweeen-sample effect. Seurat implements several such methods, and has a full <a href="https://satijalab.org/seurat/articles/integration_introduction">‘Introduction to scRNA-seq integration’ vignette</a>. We will use the ‘harmony’ method, from Korsunsky et al 2019: <a href="https://www.nature.com/articles/s41592-019-0619-0" class="uri">https://www.nature.com/articles/s41592-019-0619-0</a></p>
<p>Its worth noting that these same methods can be applied to combine data from multiple experiments (e.g. public data), or even technologies!</p>
<p>This is one of the most computationally intensive steps of the analysis. It can take hours to run and use many Gb of RAM, scaling with the size of your experiment.</p>
<p>This is the code to the integration and generate a new UMAP. Do not run this code today!</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="batch-correction.html#cb44-1" tabindex="-1"></a>  <span class="co"># https://satijalab.org/seurat/articles/seurat5_integration</span></span>
<span id="cb44-2"><a href="batch-correction.html#cb44-2" tabindex="-1"></a>  <span class="co"># Slow?</span></span>
<span id="cb44-3"><a href="batch-correction.html#cb44-3" tabindex="-1"></a>  so <span class="ot">&lt;-</span> <span class="fu">IntegrateLayers</span>(so, <span class="at">method =</span> HarmonyIntegration, <span class="at">orig.reduction =</span> <span class="st">"pca"</span>, <span class="at">new.reduction =</span> <span class="st">"harmony"</span>)</span>
<span id="cb44-4"><a href="batch-correction.html#cb44-4" tabindex="-1"></a>  so <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(so, <span class="at">dims=</span><span class="dv">1</span><span class="sc">:</span>num_dims, <span class="at">reduction =</span> <span class="st">'harmony'</span>)</span>
<span id="cb44-5"><a href="batch-correction.html#cb44-5" tabindex="-1"></a></span>
<span id="cb44-6"><a href="batch-correction.html#cb44-6" tabindex="-1"></a>  <span class="fu">DimPlot</span>(so, <span class="at">group.by=</span><span class="st">'tissue_sample'</span>)</span>
<span id="cb44-7"><a href="batch-correction.html#cb44-7" tabindex="-1"></a></span>
<span id="cb44-8"><a href="batch-correction.html#cb44-8" tabindex="-1"></a>  </span>
<span id="cb44-9"><a href="batch-correction.html#cb44-9" tabindex="-1"></a>  <span class="do">## </span><span class="al">TODO</span><span class="do"> Remove clustering from this section, update object</span></span>
<span id="cb44-10"><a href="batch-correction.html#cb44-10" tabindex="-1"></a>  <span class="co">#so &lt;- FindNeighbors(so, reduction = "harmony", dims = 1:num_dims)</span></span>
<span id="cb44-11"><a href="batch-correction.html#cb44-11" tabindex="-1"></a>  <span class="co">#so &lt;- FindClusters(so, resolution = 0.2)</span></span>
<span id="cb44-12"><a href="batch-correction.html#cb44-12" tabindex="-1"></a>  <span class="co"># don't talk about optimisation of clusters, point people at single cell resources.</span></span>
<span id="cb44-13"><a href="batch-correction.html#cb44-13" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
<p>Instead, load up a version of the object with theis data.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_file_01_preprocessed_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_01_preprocessed_subsampled.RDS"</span><span class="op">)</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">seurat_file_01_preprocessed_subset</span><span class="op">)</span></span></code></pre></div>
<p>Looking in the summary of the seurat object we see 3x dimensional reductions:</p>
<ul>
<li>
<strong>pca</strong> : Our original principal components, with no batch correction.</li>
<li>
<strong>harmony</strong> : The harmony output, values like the PCA, but adjusted for batch correction.</li>
<li>
<strong>umap</strong> : Our new umap which was based on harmony (because we passed <em>reduction = ‘harmony’</em> to RunUMAP). This overwrote the old UMAP.</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span></span></code></pre></div>
<pre><code>## An object of class Seurat 
## 999 features across 50662 samples within 2 assays 
## Active assay: RNA (980 features, 200 variable features)
##  13 layers present: counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c, counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c, data.GSM7473682.HC.a, data.GSM7473683.HC.b, data.GSM7473684.HC.c, data.GSM7473688.CD.a, data.GSM7473689.CD.b, data.GSM7473690.CD.c, scale.data
##  1 other assay present: negprobes
##  3 dimensional reductions calculated: pca, harmony, umap
##  6 spatial fields of view present: GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b GSM7473690.CD.c</code></pre>
<p>When plotting the new UMAP, the samples are move overlayed. This will make for better vizualisation.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, group.by<span class="op">=</span><span class="st">'orig.ident'</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-30-1.png" width="672"></div>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, group.by<span class="op">=</span><span class="st">'tissue_sample'</span>, split.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-31-1.png" width="672"></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="reduce-dimensionality---umaps.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></div>
<div class="next"><a href="clustering.html"><span class="header-section-number">11</span> Clustering</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#batch-correction"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="nav-link" href="#do-we-need-batch-correction"><span class="header-section-number">10.1</span> Do we need batch correction?</a></li>
<li><a class="nav-link" href="#running-batch-correction"><span class="header-section-number">10.2</span> Running batch correction</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-03.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
