<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 Input Data | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="What does the data look like when you get it? Examine raw data. Explain e.g. cosmx in detail, refer to similarities in e.g. xenium Load data into object Brief explanation of different data...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 6 Input Data | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="What does the data look like when you get it? Examine raw data. Explain e.g. cosmx in detail, refer to similarities in e.g. xenium Load data into object Brief explanation of different data...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 Input Data | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="What does the data look like when you get it? Examine raw data. Explain e.g. cosmx in detail, refer to similarities in e.g. xenium Load data into object Brief explanation of different data...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="exampleexperiment.html"><span class="header-section-number">4</span> Example Experiment</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="active" href="Data.html"><span class="header-section-number">6</span> Input Data</a></li>
<li><a class="" href="qc.html"><span class="header-section-number">7</span> QC</a></li>
<li><a class="" href="basic-visualisation.html"><span class="header-section-number">8</span> Basic visualisation</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">13</span> Spatially Restricted Genes</a></li>
<li><a class="" href="next-steps.html"><span class="header-section-number">14</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">15</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="Data" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> Input Data<a class="anchor" aria-label="anchor" href="#Data"><i class="fas fa-link"></i></a>
</h1>
<p>What does the data look like when you get it? Examine raw data.</p>
<ul>
<li>Explain e.g. cosmx in detail, refer to similarities in e.g. xenium</li>
<li>Load data into object</li>
<li>Brief explanation of different data structures</li>
<li>Sample annotation: Applying sample names to individual slides. Mention difficulty of multiple samples per slide (can we recommend some approach?)</li>
</ul>
<p>=====</p>
<div id="the-raw-data" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> The raw data<a class="anchor" aria-label="anchor" href="#the-raw-data"><i class="fas fa-link"></i></a>
</h2>
<p>For a cosmx SMI slide (like this experiment).</p>
<ul>
<li>
<strong>Flat files</strong> : Most of what we need to do processing. Typically a directory of flat files is read in as a whole, and we don’t need to worry too much. But note there are no images in here.
<ul>
<li>SLIDE-polygons.csv.gz : Cell borders</li>
<li>SLIDE_exprMat_file.csv.gz : Counts of genes per cell (Counts matrix)</li>
<li>SLIDE_fov_positions_file.csv.gz : Location of FOVs on slide</li>
<li>SLIDE_metadata_file.csv.gz : Cell level QC metadata</li>
<li>SLIDE_tx_file.csv.gz : Location of individual transcripts.</li>
</ul>
</li>
<li>
<strong>Raw files</strong> : Giant ugly directory with lots of files including microscopy images.
<ul>
<li>RawFiles/SLIDE/RUN_CODE/CellStatsDir/Morphology2D : Location of images</li>
<li><a href="https://github.com/Nanostring-Biostats/CosMxDACustomModules/blob/main/Export/CosMxDAExportSetup.docx" class="uri">https://github.com/Nanostring-Biostats/CosMxDACustomModules/blob/main/Export/CosMxDAExportSetup.docx</a></li>
</ul>
</li>
</ul>
<p>For a Xenium slide there is just the one typical output directory; they discuss the formats in the section on <a href="https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-archive-data">data archiving</a>, but again, software tools take the directory as a whole.</p>
</div>
<div id="loading-the-data" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"><i class="fas fa-link"></i></a>
</h2>
<div id="load-one-sample-from-raw-data" class="section level3" number="6.2.1">
<h3>
<span class="header-section-number">6.2.1</span> Load one sample from raw data<a class="anchor" aria-label="anchor" href="#load-one-sample-from-raw-data"><i class="fas fa-link"></i></a>
</h3>
<p>Load libraries, and paths</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## Loading required package: SeuratObject</code></pre>
<pre><code>## Loading required package: sp</code></pre>
<pre><code>## 
## Attaching package: 'SeuratObject'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, t</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.github.io/seurat-object/">SeuratObject</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.5
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ ggplot2   3.5.2     ✔ tibble    3.3.0
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.1
## ✔ purrr     1.1.0</code></pre>
<pre><code>## ── Conflicts ──────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://here.r-lib.org/">here</a></span><span class="op">)</span></span></code></pre></div>
<pre><code>## here() starts at /home/s.williams/projects/intro-spatial-transcriptomics-workshop</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Paths</span></span>
<span><span class="va">data_dir</span>              <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"data/"</span><span class="op">)</span></span>
<span><span class="va">raw_data_dir</span>          <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"raw_data/"</span><span class="op">)</span></span></code></pre></div>
<p>The code to load one sample should be as follows.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load one sample?</span></span>
<span><span class="va">sample_path</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">raw_data_dir</span>, <span class="st">"GSM7473682_HC_a"</span><span class="op">)</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ReadNanostring.html">LoadNanostring</a></span><span class="op">(</span><span class="va">sample_path</span>,</span>
<span>                     assay<span class="op">=</span><span class="st">'RNA'</span>,</span>
<span>                     fov<span class="op">=</span><span class="st">"GSM7473682_HC_a"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># HOWEVER </span></span>
<span><span class="co"># This default method drops most of the metatdata in the seurat object.</span></span>
<span><span class="co"># e.g. what fov is each cell a member of? is missing.</span></span>
<span></span>
<span><span class="co">#so@meta.data</span></span>
<span><span class="co">#orig.ident nCount_RNA nFeature_RNA</span></span>
<span><span class="co">#1_1   SeuratProject        368          189</span></span>
<span><span class="co">#2_1   SeuratProject        810          286</span></span>
<span><span class="co">#3_1   SeuratProject        119           74</span></span>
<span></span>
<span><span class="co"># An alternative function is here;</span></span>
<span><span class="co"># In time, this should be fixed within seruat</span></span>
<span><span class="co"># See comments here</span></span>
<span><span class="co">#https://github.com/satijalab/seurat/discussions/9261</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"scripts/LoadNanostring_edited_function.R"</span><span class="op">)</span></span></code></pre></div>
<p>Each sample should be annotated with its experimental details. This particular
study has one sample per slide (easy!), but there are typically more.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">$</span><span class="va">tissue_sample</span>   <span class="op">&lt;-</span> <span class="st">"HC_A"</span></span>
<span><span class="va">so</span><span class="op">$</span><span class="va">group</span>           <span class="op">&lt;-</span> <span class="st">"HC"</span></span>
<span><span class="va">so</span><span class="op">$</span><span class="va">condition</span>       <span class="op">&lt;-</span> <span class="st">"Healthy Controls"</span></span></code></pre></div>
</div>
<div id="load-all-samples-from-raw-data" class="section level3" number="6.2.2">
<h3>
<span class="header-section-number">6.2.2</span> Load all samples from raw data<a class="anchor" aria-label="anchor" href="#load-all-samples-from-raw-data"><i class="fas fa-link"></i></a>
</h3>
<p>Code example goes here (not run)</p>
</div>
<div id="load-the-workshop-data" class="section level3" number="6.2.3">
<h3>
<span class="header-section-number">6.2.3</span> Load the workshop data<a class="anchor" aria-label="anchor" href="#load-the-workshop-data"><i class="fas fa-link"></i></a>
</h3>
<p>This dataset is a subset of the experimenal data - only the first 5 fov views of each, and only the CD and HC sample groups.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data_dir</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="aims.html"><span class="header-section-number">5</span> Aims for today</a></div>
<div class="next"><a href="qc.html"><span class="header-section-number">7</span> QC</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#Data"><span class="header-section-number">6</span> Input Data</a></li>
<li><a class="nav-link" href="#the-raw-data"><span class="header-section-number">6.1</span> The raw data</a></li>
<li>
<a class="nav-link" href="#loading-the-data"><span class="header-section-number">6.2</span> Loading the data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#load-one-sample-from-raw-data"><span class="header-section-number">6.2.1</span> Load one sample from raw data</a></li>
<li><a class="nav-link" href="#load-all-samples-from-raw-data"><span class="header-section-number">6.2.2</span> Load all samples from raw data</a></li>
<li><a class="nav-link" href="#load-the-workshop-data"><span class="header-section-number">6.2.3</span> Load the workshop data</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-06.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
