<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 13 Next steps | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="We’ve done a fairly streamlined preprocessing of a single cell spatial dataset. At this point in the analysis it would be time to drill into the biology and start doing some focussed analyses....">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 13 Next steps | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="We’ve done a fairly streamlined preprocessing of a single cell spatial dataset. At this point in the analysis it would be time to drill into the biology and start doing some focussed analyses....">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 13 Next steps | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="We’ve done a fairly streamlined preprocessing of a single cell spatial dataset. At this point in the analysis it would be time to drill into the biology and start doing some focussed analyses....">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="study-background.html"><span class="header-section-number">3</span> Study background</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">4</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">5</span> Quality control and filtering</a></li>
<li><a class="" href="normalisation-and-scaling.html"><span class="header-section-number">6</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">7</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">8</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">9</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">10</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">11</span> Spatially Restricted Genes</a></li>
<li><a class="" href="Summary.html"><span class="header-section-number">12</span> Summary</a></li>
<li><a class="active" href="NextSteps.html"><span class="header-section-number">13</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">14</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="NextSteps" class="section level1" number="13">
<h1>
<span class="header-section-number">13</span> Next steps<a class="anchor" aria-label="anchor" href="#NextSteps"><i class="fas fa-link"></i></a>
</h1>
<p>We’ve done a fairly streamlined preprocessing of a single cell spatial dataset. At this point in the analysis it would be time to drill into the biology and start doing some focussed analyses. What might be possible?</p>
<div id="what-did-the-authors-find-from-this-dataset" class="section level2" number="13.1">
<h2>
<span class="header-section-number">13.1</span> What did the authors find from this dataset?<a class="anchor" aria-label="anchor" href="#what-did-the-authors-find-from-this-dataset"><i class="fas fa-link"></i></a>
</h2>
<p>In their paper <a href="https://pubmed.ncbi.nlm.nih.gov/37495570/">Macrophage and neutrophil heterogeneity at single-cell spatial resolution in human inflammatory bowel disease</a> Garrido-Trigo et al performed and extensive analysis of their cosmx Data alongside single cell, bulk and microscopy data.
To summarise the kinds of things that are possible with single-cell spatial data - here is a list of some of the analyses from that paper that specifically use this spatial dataset;</p>
<ul>
<li>Identified their detailed celltypes in their single cell data (over the whole transcriptome), which they used to classify celltypes in their spatial data. They classified their data at broad and specific levels (page 2 - <em>Integration of single-cell RNA sequencing and spatial molecular imaging analysis provides a map of healthy and inflamed colon</em>, Figs 1b and part of Supp Fig 1b for epithelia)</li>
</ul>
<p><img src="images/GTfig_1b_celltypes1.png" style="width:49.0%"><img src="images/GTfig_supp1b_celltypes2.png" style="width:49.0%"></p>
<ul>
<li>Looked at differential abundance of celltypes across samples (page 2 - <em>Transcriptional analysis at single-cell and spatial resolution reveals different populations of resident and inflammatory macrophages in the colonic mucosa</em> )</li>
<li>Identified cell niches with different neighbourhoods, e.g lymphoid structures, lamina propria and lower crypt. They quantified an increase in inflammatory celltypes in some of those neighbourhoods, and characterised another neighbourhood seen only in the IBD samples. (page 2 - <em>Integration of single-cell RNA sequencing and spatial molecular imaging analysis provides a map of healthy and inflamed colon</em>)</li>
</ul>
<div class="inline-figure"><img src="images/GTFig2g_niche.png" style="width:100.0%"></div>
<ul>
<li>They observed granuloma features in one patient. They examined the macrophage subtypes within them and the surrounding immune cell microenvironment. (Figure 4b, page 6 <em>CosMx Spatial Molecular Imaging analysis confirms the expansion of Inflammation-Dependent Alternative macrophages and reveals their tissue distribution in inflammatory bowel disease colon</em>)</li>
</ul>
<div class="inline-figure"><img src="images/GTfig_supp9ab_granuloma.png" style="width:100.0%"></div>
<ul>
<li>Identified a change in myeloid cell localisation by quantifying the distance of macrophage, MAST cells and Dendritic cells to the mucosal surface (page 2 - <em>Transcriptional analysis at single-cell and spatial resolution reveals different populations of resident and inflammatory macrophages in the colonic mucosa</em> )</li>
<li>Observed spatially localised expression of a gene (NRG1) within a subset of apically-located fibroblasts (page 4 - <em>Inflammation-dependent alternative macrophages express neuregulin 1</em>)</li>
</ul>
</div>
<div id="OtherTechniques" class="section level2" number="13.2">
<h2>
<span class="header-section-number">13.2</span> Other techniques<a class="anchor" aria-label="anchor" href="#OtherTechniques"><i class="fas fa-link"></i></a>
</h2>
<div id="imaging-related" class="section level3" number="13.2.1">
<h3>
<span class="header-section-number">13.2.1</span> Imaging related<a class="anchor" aria-label="anchor" href="#imaging-related"><i class="fas fa-link"></i></a>
</h3>
<p><strong>Segmentation</strong></p>
<p>Good cell segmentation from images underpins downstream analysis. Different platforms offer inbuilt segmentation methods, and often require tweaking from the default with different tissues, antibodies or runs.</p>
<ul>
<li>On-machine segmentation: e.g. Xenium/CosMx</li>
<li>Directly from images using specialised microscopy methods for segmentation. e.g. CellPose</li>
<li>Many more resources online…</li>
</ul>
<p><strong>Image analysis</strong></p>
<ul>
<li>
<a href="https://cellprofiler.org/" class="uri">https://cellprofiler.org/</a> : One toolkit often used in traditional imaging for various tasks, e.g. cell morphology quantifications.</li>
<li>Many more resources online…</li>
</ul>
<p><strong>Image registration </strong></p>
<p>Especially for multiomics analysis, being able to align images of the same tissue from different sources is essential. E.g. H&amp;E, immunofluorescence, proteomics e.t.c</p>
<ul>
<li>
<a href="https://lmweber.org/OSTA/pages/crs-spatial-registration.html" class="uri">https://lmweber.org/OSTA/pages/crs-spatial-registration.html</a> : Image registration, for aligning images from different sources. E.g. H&amp;E, immunofluorescence, proteomics e.t.c</li>
<li>Many more resources online…</li>
</ul>
<p><strong>Sample and Regional Annotation</strong></p>
<p>It’s typical to have multiple samples on a single slide. We need to label each cell with its ID. Similarly you might want to outline a particular region.</p>
<ul>
<li>Define samples by their centroid x-y coordinates (Painstaking, but it can be enough)</li>
<li>Vendor-based solutions; e.g. regional highlighting with <a href="https://www.10xgenomics.com/support/software/xenium-explorer/latest/tutorials/xe-selecting-multiple-regions-of-interest">Xenium explorer</a>
</li>
<li>Import masks from annotated images</li>
<li>Do you know better approaches or tools? Please share.</li>
</ul>
</div>
<div id="downtream-annotation-and-analyses" class="section level3" number="13.2.2">
<h3>
<span class="header-section-number">13.2.2</span> Downtream annotation and analyses<a class="anchor" aria-label="anchor" href="#downtream-annotation-and-analyses"><i class="fas fa-link"></i></a>
</h3>
<p><strong>Niche analysis</strong></p>
<p>Cell ‘niche’ analysis or ‘spatial context’ is a major avenue of in situ spatial analyses. E.g. tumour microenvrionments, tissue regions. The actual definition of what a ‘niche’ is could be considered flexible - it depends on what you are trying to achieve - and different tools take differnt approches.</p>
<ul>
<li>Seurat findNiches()</li>
<li>OSCA neighbourhood analyses: <a href="https://lmweber.org/OSTA/pages/img-neighbourhood-analysis.html" class="uri">https://lmweber.org/OSTA/pages/img-neighbourhood-analysis.html</a>
</li>
<li>hoodscanR : <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/hoodscanR/inst/doc/Quick_start.html" class="uri">https://www.bioconductor.org/packages/release/bioc/vignettes/hoodscanR/inst/doc/Quick_start.html</a>
</li>
<li>GraphST : <a href="https://www.nature.com/articles/s41467-023-36796-3" class="uri">https://www.nature.com/articles/s41467-023-36796-3</a>
</li>
<li>Many more …</li>
</ul>
<p><strong>‘Spatial’ pattern tests and cell free analyses</strong></p>
<p>Huge area of research, here are some starting points;</p>
<ul>
<li>Co-localisation and differential co-localisation: <a href="https://lmweber.org/OSTA/pages/mult-diff-spatial-patterns.html" class="uri">https://lmweber.org/OSTA/pages/mult-diff-spatial-patterns.html</a>
</li>
<li>Statial toolkit (Kontextual and Spatiomark) for quantifying spatial relationhips and expression with spatial contet: <a href="https://bioconductor.org/packages/release/bioc/html/Statial.html">Statial package</a>
</li>
<li>Spatial tests, adapted form geospatial methods: <a href="https://pachterlab.github.io/voyager/" class="uri">https://pachterlab.github.io/voyager/</a>
</li>
<li>Molecule level data object (not a SingleCellExperiment): <a href="https://www.bioconductor.org/packages/release/bioc/html/MoleculeExperiment.html" class="uri">https://www.bioconductor.org/packages/release/bioc/html/MoleculeExperiment.html</a>
</li>
<li>Many more …</li>
</ul>
<p><strong>Differential expression</strong></p>
<p>Though current gen single-cell spatial techonlogies have lower counts and fewer genes in a panel than single cell (and bulk) studies, being able to test for differences in gene expression is valuable; potentially within a celltype, niche or spatial region that couldn’t be isolated with other technologies. With this in mind, single cell methods can be applied.</p>
<ul>
<li>Differential expression in scRNAseq (bioconductor): <a href="https://bioconductor.org/books/3.21/OSCA.multisample/multi-sample-comparisons.html" class="uri">https://bioconductor.org/books/3.21/OSCA.multisample/multi-sample-comparisons.html</a>
</li>
<li>Applied example (in this same dataset!) running differential expresison at the celltype level using <a href="https://swbioinf.github.io/spatialsnippets/e_DEPseudobulkSampleLevel_insitu.html#Overview">Seurat</a> or <a href="https://swbioinf.github.io/spatialsnippets/e_DEPseudobulk_insitu_sfe.html">bioconductor ecosystem</a>
</li>
<li>Spatially-aware differential expression methods; These are emerging quickly, consider looking into <a href="https://www.nature.com/articles/s41598-024-61758-0">Differential gene expression analysis of spatial transcriptomic experiments using spatial mixed models</a> and <a href="https://github.com/dmcable/spacexr">spacexr</a>
</li>
<li>Many more …</li>
</ul>
<p><strong>Differential abundance</strong></p>
<p>Differences in the numbers or types of cell in a particular location or neighbour niche between groups can indicate biological differences; from cell migration, expansion or development.</p>
<ul>
<li>Propeller for testing celltype proportion changes: <a href="https://github.com/phipsonlab/speckle" class="uri">https://github.com/phipsonlab/speckle</a>
</li>
<li>Many more …</li>
</ul>
<p><strong>Multiomics</strong></p>
<p>Another huge area of research, here are some starting points. It very much depends on which multiomics you have.</p>
<ul>
<li>Seurat multimodal vignette (single cell): <a href="https://satijalab.org/seurat/articles/multimodal_vignette" class="uri">https://satijalab.org/seurat/articles/multimodal_vignette</a> (single cell)</li>
<li>Image registration, for aligning images from different sources. E.g. H&amp;E, immunofluorescence, proteomics e.t.c : <a href="https://lmweber.org/OSTA/pages/crs-spatial-registration.html" class="uri">https://lmweber.org/OSTA/pages/crs-spatial-registration.html</a>
</li>
<li>Many more …</li>
</ul>
</div>
<div id="visualisation" class="section level3" number="13.2.3">
<h3>
<span class="header-section-number">13.2.3</span> Visualisation<a class="anchor" aria-label="anchor" href="#visualisation"><i class="fas fa-link"></i></a>
</h3>
<p>There are many tools out there, each with strengths and limitations.</p>
<ul>
<li>VRomics: <a href="https://ramialison-lab.github.io/pages/vromics.html" class="uri">https://ramialison-lab.github.io/pages/vromics.html</a>
</li>
<li>Xenium explorer (Xenium only): <a href="https://www.10xgenomics.com/support/software/xenium-explorer/latest" class="uri">https://www.10xgenomics.com/support/software/xenium-explorer/latest</a>
</li>
<li>AtoMx (cosmx only) : <a href="https://nanostring.com/products/atomx-spatial-informatics-platform/atomx-sip-overview/" class="uri">https://nanostring.com/products/atomx-spatial-informatics-platform/atomx-sip-overview/</a>
</li>
<li>Napari <a href="https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/posts/napari-cosmx-basics/using-napari-cosmx.html">cosmx guide</a>
</li>
<li>CellXGene (single cell) : <a href="https://cellxgene.cziscience.com/" class="uri">https://cellxgene.cziscience.com/</a>
</li>
<li>iSEE (single cell): <a href="https://github.com/iSEE/iSEE" class="uri">https://github.com/iSEE/iSEE</a>
</li>
<li>Many more …</li>
</ul>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="Summary.html"><span class="header-section-number">12</span> Summary</a></div>
<div class="next"><a href="resources.html"><span class="header-section-number">14</span> Resources</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#NextSteps"><span class="header-section-number">13</span> Next steps</a></li>
<li><a class="nav-link" href="#what-did-the-authors-find-from-this-dataset"><span class="header-section-number">13.1</span> What did the authors find from this dataset?</a></li>
<li>
<a class="nav-link" href="#OtherTechniques"><span class="header-section-number">13.2</span> Other techniques</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#imaging-related"><span class="header-section-number">13.2.1</span> Imaging related</a></li>
<li><a class="nav-link" href="#downtream-annotation-and-analyses"><span class="header-section-number">13.2.2</span> Downtream annotation and analyses</a></li>
<li><a class="nav-link" href="#visualisation"><span class="header-section-number">13.2.3</span> Visualisation</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-29.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
