<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 Normalisation and scaling | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="6.1 Goals Correct for biases across cells and gene expression values Understand the trade-offs between normalisation methods Identify the most variable genes in preparation for dimensionality...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 6 Normalisation and scaling | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="6.1 Goals Correct for biases across cells and gene expression values Understand the trade-offs between normalisation methods Identify the most variable genes in preparation for dimensionality...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 Normalisation and scaling | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="6.1 Goals Correct for biases across cells and gene expression values Understand the trade-offs between normalisation methods Identify the most variable genes in preparation for dimensionality...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="study-background.html"><span class="header-section-number">3</span> Study background</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">4</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">5</span> Quality control and filtering</a></li>
<li><a class="active" href="normalisation-and-scaling.html"><span class="header-section-number">6</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">7</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">8</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">9</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">10</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">11</span> Spatially Restricted Genes</a></li>
<li><a class="" href="Summary.html"><span class="header-section-number">12</span> Summary</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">13</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">14</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="normalisation-and-scaling" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> Normalisation and scaling<a class="anchor" aria-label="anchor" href="#normalisation-and-scaling"><i class="fas fa-link"></i></a>
</h1>
<div class="inline-figure"><img src="images/workflows/03_normalisation.png" style="width:100.0%"></div>
<div id="goals-3" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> Goals<a class="anchor" aria-label="anchor" href="#goals-3"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Correct for biases across cells and gene expression values</li>
<li>Understand the trade-offs between normalisation methods</li>
<li>Identify the most variable genes in preparation for dimensionality reduction</li>
</ul>
</div>
<div id="overview-3" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> Overview<a class="anchor" aria-label="anchor" href="#overview-3"><i class="fas fa-link"></i></a>
</h2>
<p>Normalisation and scaling are important processing steps when analysing
transcriptomics data, particulary for single-cell RNA sequencing (scRNA-seq)
and spatial transcriptomics (ST). They serve different purposes but both aim
to adjust technical variation so that cells are comparable. This is important for accurate clustering and analysis tasks.</p>
<p>Goal: capture biology.</p>
<ol style="list-style-type: decimal">
<li>
<strong>Normalisation:</strong> Different cells have different total target counts. Normalisation allows them to be comparable.<br>
</li>
<li>
<strong>Identification of highly variable features (feature selection)</strong> Identifying the most variable features allows retaining the real biological variability of the data and reduce noise in the data.<br>
</li>
<li>
<strong>Scaling the data</strong>: Highly expressed genes can overpower the signal of other less expressed genes with equal importance. Within the same cell the assumption is that the underlying RNA content is constant.</li>
</ol>
<div class="inline-figure"><img src="images/normandscale.png" style="width:100.0%"></div>
</div>
<div id="selecing-a-normalisation-approach" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> Selecing a normalisation approach<a class="anchor" aria-label="anchor" href="#selecing-a-normalisation-approach"><i class="fas fa-link"></i></a>
</h2>
<p>The choice of normalisation method can have a big impact on the biological
interpretation of your downstream analysis steps. Therefore, it is important
to consider selecting a suitable approach based on your data, and the questions
you want to answer.</p>
<p>Unlike scRNA-seq, where cells are treated as independent observations, ST data
have spatial patterns that need to be accounted for that scRNA-seq
normalisation misses. These patterns include probe capture performance,
differences in cell morphology and heterogeneous tissue architecture.</p>
<p>Other sources include gene panel skew.</p>
<div class="rmdimportant">
<p><strong>Recommendations</strong></p>
<p>Trade-off between simplicity and performance of methods.</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="15%">
<col width="32%">
<col width="52%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>Pros</th>
<th>Cons</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Spatially-aware (SpaNorm)</td>
<td>Recommended - accounts for spatial dependent patterns</td>
<td>Computationally demanding and requires several parameter tuning steps</td>
</tr>
<tr class="even">
<td>Cell volume/area normalisation</td>
<td>Alternative for spatially-aware methods</td>
<td>Requires accurate area or volume data, bad cell segmentation = bad normalisation, manual</td>
</tr>
<tr class="odd">
<td>Log-normalisation</td>
<td>Easy to run, incorporated in many intermediate processing pipelines, used for analyses that require “minimal” normalisation</td>
<td>Does not account for spatial patterns</td>
</tr>
<tr class="even">
<td>scTransform</td>
<td>Powerful for scRNA-seq</td>
<td>Avoid for ST - over-corrects spatial patterns</td>
</tr>
<tr class="odd">
<td>Library size normalisation</td>
<td>Simple</td>
<td>Avoid, problematic overall</td>
</tr>
</tbody>
</table></div>
<p>For a more in-depth discussion of normalisation in ST data sets and benchmarks,
refer to the following resources:</p>
<ul>
<li>
<a href="https://lmweber.org/OSTA/pages/ind-normalization.html">OSTA</a>
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-024-03303-w">Atta et al. (2024)</a>
</li>
<li><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-025-03565-y">Bhuva et al. (2024)</a></li>
</ul>
</div>
<!-- CHCEK IF FINAL OBJEWCT IS SPLIT BY NOW -- Start by 'splitting' the object so that each sample has its own assay again.  -->
<!-- ```{r  results='hide', warnings=FALSE} -->
<!-- # Basic preprocessing -->
<!-- # Split layers out again -->
<!-- # https://satijalab.org/seurat/articles/seurat5_integration -->
<!-- #so <- split(so, f = so$orig.ident) -->
<!-- ``` -->
<p>Inspect current RNA assay. Layers are <code>counts.*</code> for each sample/FOV.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span></span></code></pre></div>
<pre><code>## Assay (v5) data with 980 features for 56853 cells
## First 10 features:
##  TGFB3, PIGR, MALAT1, MZT2A, IL16, XBP1, S100A6, CCL2, CEACAM1, VSIR 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c</code></pre>
<div class="rmdimportant">
<p><strong>Choosing a normalisation method is context-dependent</strong></p>
<p>Normalise:</p>
<ul>
<li>For dimensionality reduction and clustering<br>
</li>
<li>Integration and batch correction<br>
</li>
<li>Cell annotation, label transfer</li>
</ul>
<p><strong>Minimally normalised</strong> counts:</p>
<p>Where raw transcript counts are required or apply their own
normalisation method. Used to preserve spatial trends
that normalisation can remove e.g.</p>
<ul>
<li>Identifying spatially variable genes - Moran’s I</li>
<li>Reporting gene expression values across cells</li>
</ul>
<p>This is why it is important to store both the <code>counts</code> and <code>data</code>
layers in your data object.</p>
</div>
<p>As we will be identifying spatially variable genes with Moran’s I, we will
apply a “minimally normalised” approach - the ‘LogNormalize’
method in Seurat.</p>
<p>As described in the NormalizeData function help:</p>
<blockquote>
<p>“LogNormalize: Feature counts for each cell are divided by the total counts for that cell and multiplied by the scale.factor. This is then natural-log transformed using log1’”</p>
</blockquote>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="va">so</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span><span class="op">)</span></span></code></pre></div>
<p>When displaying the RNA counts assay again, note that there are new
<code>data</code> layers. These represent our (log-)normalised data.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span></span></code></pre></div>
<pre><code>## Assay (v5) data with 980 features for 56853 cells
## First 10 features:
##  TGFB3, PIGR, MALAT1, MZT2A, IL16, XBP1, S100A6, CCL2, CEACAM1, VSIR 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c,
## data.GSM7473682.HC.a, data.GSM7473683.HC.b, data.GSM7473684.HC.c,
## data.GSM7473688.CD.a, data.GSM7473689.CD.b, data.GSM7473690.CD.c</code></pre>
</div>
<div id="identifying-highly-variable-genes-hvgs" class="section level2" number="6.4">
<h2>
<span class="header-section-number">6.4</span> Identifying highly variable genes (HVGs)<a class="anchor" aria-label="anchor" href="#identifying-highly-variable-genes-hvgs"><i class="fas fa-link"></i></a>
</h2>
<div class="inline-figure"><img src="images/genes2umap.png" style="width:100.0%"></div>
<p>Which genes vary between cell types/states in our sample?</p>
<p>We don’t yet know anything about celltypes or stats, but we can make the assumption that the genes with high variance and are probably reflecting some biological state. In contrast a ‘housekeeper’ gene with even expression across all cell types won’t be helpful in grouping our cells.</p>
<p>Because we have our data split by sample, Seurat will identify variable genes within each sample and combine that information. From the <a href="https://satijalab.org/seurat/articles/seurat5_integration#layers-in-the-seurat-v5-object">Seurat5 documentation on layers</a>:</p>
<blockquote>
<p>“Note that since the data is split into layers, normalization and variable feature identification is performed for each batch independently (a consensus set of variable features is automatically identified).”</p>
</blockquote>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">so</span>, nfeatures <span class="op">=</span> <span class="fl">200</span><span class="op">)</span></span></code></pre></div>
<p>Each point represents one gene - the red ones are our top 200 ‘highly variable genes’ or HVGs. There is no firm rule on how many HVGs we want. For a whole transcriptome that could be 2000, for a 1000 genes we have 200. The exact number doesn’t matter so much; so long as we capture the genes above the main group of low variance genes.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/VariableFeaturePlot.html">VariableFeaturePlot</a></span><span class="op">(</span><span class="va">so</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-46-1.png" width="672"></div>
<p>We can list those genes:</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span><span class="va">so</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>## [1] "IGHM"   "IGHG1"  "TPSB2"  "IGHG2"  "JCHAIN" "TPSAB1"</code></pre>
</div>
<div id="scale-data" class="section level2" number="6.5">
<h2>
<span class="header-section-number">6.5</span> Scale data<a class="anchor" aria-label="anchor" href="#scale-data"><i class="fas fa-link"></i></a>
</h2>
<p>By default, the scaleData function will scale only the HVGs.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html">ScaleData</a></span><span class="op">(</span><span class="va">so</span><span class="op">)</span> <span class="co"># Just 200 variable features</span></span></code></pre></div>
<p>Note the presence of the single ‘scale.data’ layer.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span></span></code></pre></div>
<pre><code>## An object of class Seurat 
## 999 features across 56853 samples within 2 assays 
## Active assay: RNA (980 features, 200 variable features)
##  13 layers present: counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c, counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c, data.GSM7473682.HC.a, data.GSM7473683.HC.b, data.GSM7473684.HC.c, data.GSM7473688.CD.a, data.GSM7473689.CD.b, data.GSM7473690.CD.c, scale.data
##  1 other assay present: negprobes
##  6 spatial fields of view present: GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b GSM7473690.CD.c</code></pre>
<p>Save the object</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">so</span>, file <span class="op">=</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_03_normalised.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="quality-control-and-filtering.html"><span class="header-section-number">5</span> Quality control and filtering</a></div>
<div class="next"><a href="ReducedDimensionality.html"><span class="header-section-number">7</span> Reduce dimensionality - UMAPs</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#normalisation-and-scaling"><span class="header-section-number">6</span> Normalisation and scaling</a></li>
<li><a class="nav-link" href="#goals-3"><span class="header-section-number">6.1</span> Goals</a></li>
<li><a class="nav-link" href="#overview-3"><span class="header-section-number">6.2</span> Overview</a></li>
<li><a class="nav-link" href="#selecing-a-normalisation-approach"><span class="header-section-number">6.3</span> Selecing a normalisation approach</a></li>
<li><a class="nav-link" href="#identifying-highly-variable-genes-hvgs"><span class="header-section-number">6.4</span> Identifying highly variable genes (HVGs)</a></li>
<li><a class="nav-link" href="#scale-data"><span class="header-section-number">6.5</span> Scale data</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-29.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
