<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 Quality control and filtering | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="7.1 Overview Basic QC plots + filtering (20min) (Negative probes if using cosmx) Identify low quality cells Read in subsampled seurat object  7.2 Identifying potential technical artefacts so_path...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 7 Quality control and filtering | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="7.1 Overview Basic QC plots + filtering (20min) (Negative probes if using cosmx) Identify low quality cells Read in subsampled seurat object  7.2 Identifying potential technical artefacts so_path...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 Quality control and filtering | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="7.1 Overview Basic QC plots + filtering (20min) (Negative probes if using cosmx) Identify low quality cells Read in subsampled seurat object  7.2 Identifying potential technical artefacts so_path...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="exampleexperiment.html"><span class="header-section-number">4</span> Example Experiment</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="active" href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="" href="basic-visualisation.html"><span class="header-section-number">8</span> Basic visualisation</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">13</span> Spatially Restricted Genes</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">14</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">15</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="quality-control-and-filtering" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> Quality control and filtering<a class="anchor" aria-label="anchor" href="#quality-control-and-filtering"><i class="fas fa-link"></i></a>
</h1>
<div id="overview-2" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> Overview<a class="anchor" aria-label="anchor" href="#overview-2"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Basic QC plots + filtering (20min)</li>
<li>(Negative probes if using cosmx)</li>
<li>Identify low quality cells</li>
</ul>
<p>Read in subsampled seurat object</p>
</div>
<div id="identifying-potential-technical-artefacts" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> Identifying potential technical artefacts<a class="anchor" aria-label="anchor" href="#identifying-potential-technical-artefacts"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so_path</span>  <span class="op">&lt;-</span> <span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">so_path</span><span class="op">)</span></span></code></pre></div>
<p>Library size (total transcripts) and cell areas are good indicators of technical
artefacts introduced during data generation steps.</p>
<p>Possible artefacts:
- low quality cells
- cell segmentation challenges (combining two cells, not aligned with real cell
boundaries, leading to high area and low RNA count)</p>
<p>Understanding these can help with biological interpretation downstream,
for example, discerning whether differences in cells are due to e.g. cell type
differences, or technical artefacts.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Total counts per cell</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">nCount_RNA</span>, col<span class="op">=</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_density</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotation_logticks</span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Transcript counts per cell"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-26-1.png" width="672"></div>
<p>The transcripts per cell all display a log-normal distribution with shifting medians.</p>
<p>Indicates no largely low-quality cells, and a good example of library size
effects that should be normalised downstream, prior to analyses.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Total counts per cell</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">Area</span>, col<span class="op">=</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_density</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotation_logticks</span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Area (px) per cell"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-27-1.png" width="672"></div>
<p>Areas also follow log-normal distributions with medians ranging between 3,000 -
4,000px^2. Sample/FOV <code>CD.c</code> has slightly smaller cell area with a median
of ~2,000px^2.</p>
<p>Useful to view bivariate metrics - the larger the cell, the more transcripts, in
general.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">Area</span>, y<span class="op">=</span><span class="va">nCount_RNA</span>, colour<span class="op">=</span><span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotation_logticks</span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"bl"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">orig.ident</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotation_logticks</span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"lightgrey"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-28-1.png" width="672"></div>
<p>Another good check for low-quality cells is to view the bivariate distribution
of transcript counts per cell vs. how many of these have been mapped successfully
to a gene.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">nCount_RNA</span>, y <span class="op">=</span> <span class="va">nFeature_RNA</span>, colour <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.4</span>, size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-29-1.png" width="672"></div>
<p>The more transcripts detected <code>nCount_RNA</code>, the more mapped genes there should be.</p>
<p>Point out where bad quality cells would reside</p>
</div>
<div id="filtering-low-quality-cells" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> Filtering low-quality cells<a class="anchor" aria-label="anchor" href="#filtering-low-quality-cells"><i class="fas fa-link"></i></a>
</h2>
<div id="negative-probes-as-an-indicator-of-noise" class="section level3" number="7.3.1">
<h3>
<span class="header-section-number">7.3.1</span> Negative probes as an indicator of noise<a class="anchor" aria-label="anchor" href="#negative-probes-as-an-indicator-of-noise"><i class="fas fa-link"></i></a>
</h3>
<p>Spatial transcriptomics assays include both RNA-specific probes, and negative
control probes (or, background probes). While RNA specific-probes hybridise to
specific targets, negative probes do not bind to anything. Their primary purpose
is to quantify and account for this technical noise and non-specific signal.</p>
<p>In this step, we will calculate the proportion of negative control probes
relative to the total counts (RNA) for each cell. Cells exhibiting a high
percentage of noise (i.e., high proportion of negative probe counts) can be
flagged and removed during the QC stage.</p>
<p>Key metrics to add to the Seurat object:</p>
<ul>
<li>
<code>so$percent_neg</code>: The percentage of negative probes per cell</li>
<li>
<code>so$mean_neg</code>: The average negative probes per cell</li>
</ul>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">$</span><span class="va">percent_neg</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">nCount_negprobes</span> <span class="op">/</span> <span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">nCount_RNA</span> <span class="op">+</span> <span class="va">so</span><span class="op">$</span><span class="va">nCount_negprobes</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span></span>
<span></span>
<span><span class="co"># Join layers to calculate average_neg</span></span>
<span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span> <span class="op">&lt;-</span> <span class="fu">JoinLayers</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">)</span></span>
<span><span class="va">so</span><span class="op">$</span><span class="va">mean_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">)</span> <span class="co"># Only defined first sample</span></span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">percent_neg</span>, colour <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_density</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotation_logticks</span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span>, colour <span class="op">=</span> <span class="st">"lightgrey"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-31-1.png" width="672"></div>
<p>Log-normally distributed. CD_b, HC_B samples are less “noisy”.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue_sample</span>, y <span class="op">=</span> <span class="va">mean_neg</span>, colour <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_col</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-32-1.png" width="672"></div>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Discuss difference with Xenium - xenium lower counts, lower background.</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-33-1.png" width="672"></div>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">coord_cartesian</span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-33-2.png" width="672"></div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_ncount_rna</span> <span class="op">&lt;-</span> <span class="fl">90</span></span>
<span><span class="va">max_percent_neg</span> <span class="op">&lt;-</span> <span class="fl">2</span></span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">max_percent_neg</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">min_ncount_rna</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">coord_cartesian</span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">21</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-35-1.png" width="672"></div>
</div>
<div id="identifying-filtering-thresholds" class="section level3" number="7.3.2">
<h3>
<span class="header-section-number">7.3.2</span> Identifying filtering thresholds<a class="anchor" aria-label="anchor" href="#identifying-filtering-thresholds"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply a filter</span></span>
<span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    qc <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">nCount_RNA</span> <span class="op">&gt;=</span> <span class="va">min_ncount_rna</span> <span class="op">&amp;</span> <span class="va">percent_neg</span> <span class="op">&lt;=</span> <span class="va">max_percent_neg</span>, <span class="st">"Keep"</span>, <span class="st">"Remove"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">qc</span>, <span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>##         
##          GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a
##   Keep              6811           14002            5326            4216
##   Remove            1984            1764            4393            3323
##         
##          GSM7473689.CD.b GSM7473690.CD.c
##   Keep             12479            8636
##   Remove             720            4970</code></pre>
<p>Nearly 50% of the cells for HC.c will be filtered out, display which cells will
be removed.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ImageDimPlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473684.HC.c"</span>, group.by <span class="op">=</span> <span class="st">"qc"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-37-1.png" width="672"></div>
<p>The stromal compartment will be removed entirely and can be difficult for
co-localisation analyses. Further, the epithelial compartment in the bottom left
FOV is very patchy.</p>
</div>
</div>
<div id="exercise-identifying-your-own-filtering-thresholds" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> Exercise: Identifying your own filtering thresholds<a class="anchor" aria-label="anchor" href="#exercise-identifying-your-own-filtering-thresholds"><i class="fas fa-link"></i></a>
</h2>
<p>You have seen an example where stringent filtering omits key cells or
compartments.</p>
<p>Identify a suitable threshold for <code>min_ncount_rna</code> and <code>max_percent_neg</code> by
iteratively:</p>
<ol style="list-style-type: decimal">
<li>Changing the thresholds for <code>min_ncount_rna</code> and <code>max_percent_neg</code>
</li>
<li>Displaying the bivariate distribution of <code>nCount_RNA</code> and <code>percent_neg</code> per
tissue sample</li>
<li>Using <code>ImageDimPlot</code> to visualise which cells are removed</li>
</ol>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_ncount_rna</span> <span class="op">&lt;-</span> <span class="fl">70</span></span>
<span><span class="va">max_percent_neg</span> <span class="op">&lt;-</span> <span class="fl">4</span></span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu">aes</span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">max_percent_neg</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">min_ncount_rna</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">coord_cartesian</span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">21</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-39-1.png" width="672"></div>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    qc <span class="op">=</span> <span class="fu">if_else</span><span class="op">(</span><span class="va">nCount_RNA</span> <span class="op">&gt;=</span> <span class="va">min_ncount_rna</span> <span class="op">&amp;</span> <span class="va">percent_neg</span> <span class="op">&lt;=</span> <span class="va">max_percent_neg</span>, <span class="st">"Keep"</span>, <span class="st">"Remove"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">qc</span>, <span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>##         
##          GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a
##   Keep              7411           14774            6456            5070
##   Remove            1384             992            3263            2469
##         
##          GSM7473689.CD.b GSM7473690.CD.c
##   Keep             12839           10303
##   Remove             360            3303</code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Epithelial compartment less patchy, but still hard to retain the stromal cells</span></span>
<span><span class="co"># These are the trade-offs to consider</span></span>
<span><span class="fu">ImageDimPlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473684.HC.c"</span>, group.by <span class="op">=</span> <span class="st">"qc"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-41-1.png" width="672"></div>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A lot of cells marked for removal and a bit patchy</span></span>
<span><span class="fu">ImageDimPlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473688.CD.a"</span>, group.by <span class="op">=</span> <span class="st">"qc"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-42-1.png" width="672"></div>
<div id="applying-the-filter" class="section level3" number="7.4.1">
<h3>
<span class="header-section-number">7.4.1</span> Applying the filter<a class="anchor" aria-label="anchor" href="#applying-the-filter"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">so</span>, <span class="va">qc</span> <span class="op">==</span> <span class="st">"Keep"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so_filtered</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b 
##            7411           14774            6456            5070           12839 
## GSM7473690.CD.c 
##           10303</code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">so_filtered</span>, file <span class="op">=</span> <span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_02_rna70_neg4.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div id="field-of-view-qc-cosmx-only-advanced" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> Field of View QC (CosMX only; Advanced)<a class="anchor" aria-label="anchor" href="#field-of-view-qc-cosmx-only-advanced"><i class="fas fa-link"></i></a>
</h2>
<p>Good to check library sizes across FOVs - red flags include FOVs with close to
no cells.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">unite</span><span class="op">(</span>col <span class="op">=</span> <span class="va">unique_fov_names</span>, <span class="va">orig.ident</span>, <span class="va">fov</span>, remove <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">nCount_RNA</span>, col <span class="op">=</span> <span class="va">orig.ident</span>, group <span class="op">=</span> <span class="va">unique_fov_names</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_density</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotation_logticks</span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-45-1.png" width="672"></div>
<p>TODO: As well as border effects. Run per sample.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For each cell, calculate the distance between the centroid to each FOV border</span></span>
<span><span class="va">so_borders</span> <span class="op">&lt;-</span> <span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">tissue_sample</span>, <span class="va">fov</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    top <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">CenterY_local_px</span><span class="op">)</span> <span class="op">-</span> <span class="va">CenterY_local_px</span>,</span>
<span>    bottom <span class="op">=</span> <span class="va">CenterY_local_px</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">CenterY_local_px</span><span class="op">)</span>,</span>
<span>    left <span class="op">=</span> <span class="va">CenterX_local_px</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">CenterX_local_px</span><span class="op">)</span>,</span>
<span>    right <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">CenterX_local_px</span><span class="op">)</span> <span class="op">-</span> <span class="va">CenterX_local_px</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">top</span>, <span class="va">bottom</span>, <span class="va">left</span>, <span class="va">right</span>, <span class="va">nCount_RNA</span>, <span class="va">tissue_sample</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">nCount_RNA</span>, <span class="va">tissue_sample</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"fov_direction"</span>, values_to <span class="op">=</span> <span class="st">"distance"</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">so_borders</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">distance</span>, y <span class="op">=</span> <span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">fov_direction</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_y_log10</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_light</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">"loess"</span>, span <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="summary" class="section level2" number="7.6">
<h2>
<span class="header-section-number">7.6</span> Summary<a class="anchor" aria-label="anchor" href="#summary"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>QC is not prescriptive</li>
<li>Explore your data and ITERATE</li>
<li>Be permissive, especially if it’s your first look at the data</li>
<li>There are more sources of technical variation in spatial, in comparison to older techs. While you don’t need to filter in every step, it is useful to understand these sources of variation to interpret your biological analyses, and inform the subsequent rounds of QC. These differ across techs. For example, CosMX have FOVs which adds another dimension in comparison to Xenium</li>
</ul>
<p>Notes:</p>
<p>You can also conduct QC in low dimensional steps following the
normalisation and dimension reduction steps in this workshop. However,
we will proceed directly with the pre-processing steps.</p>
<p>Also you can consider filtering each sample separately. Applying a single, strict threshold based on a high-quality sample might inappropriately filter out valid cells from a technically lower-quality sample.
- TODO: fact check</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></div>
<div class="next"><a href="basic-visualisation.html"><span class="header-section-number">8</span> Basic visualisation</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#quality-control-and-filtering"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="nav-link" href="#overview-2"><span class="header-section-number">7.1</span> Overview</a></li>
<li><a class="nav-link" href="#identifying-potential-technical-artefacts"><span class="header-section-number">7.2</span> Identifying potential technical artefacts</a></li>
<li>
<a class="nav-link" href="#filtering-low-quality-cells"><span class="header-section-number">7.3</span> Filtering low-quality cells</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#negative-probes-as-an-indicator-of-noise"><span class="header-section-number">7.3.1</span> Negative probes as an indicator of noise</a></li>
<li><a class="nav-link" href="#identifying-filtering-thresholds"><span class="header-section-number">7.3.2</span> Identifying filtering thresholds</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#exercise-identifying-your-own-filtering-thresholds"><span class="header-section-number">7.4</span> Exercise: Identifying your own filtering thresholds</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#applying-the-filter"><span class="header-section-number">7.4.1</span> Applying the filter</a></li></ul>
</li>
<li><a class="nav-link" href="#field-of-view-qc-cosmx-only-advanced"><span class="header-section-number">7.5</span> Field of View QC (CosMX only; Advanced)</a></li>
<li><a class="nav-link" href="#summary"><span class="header-section-number">7.6</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-08.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
