<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 Quality control and filtering | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="7.1 Learning objectives Recall common quality control (QC) metrics used in spatial transcriptomics, such as total transcript counts, detected genes, cell area, and negative probe percentage Recall...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 7 Quality control and filtering | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="7.1 Learning objectives Recall common quality control (QC) metrics used in spatial transcriptomics, such as total transcript counts, detected genes, cell area, and negative probe percentage Recall...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 Quality control and filtering | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="7.1 Learning objectives Recall common quality control (QC) metrics used in spatial transcriptomics, such as total transcript counts, detected genes, cell area, and negative probe percentage Recall...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="study-overview.html"><span class="header-section-number">4</span> Study overview</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="active" href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="" href="normalisation-and-scaling.html"><span class="header-section-number">8</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">13</span> Spatially Restricted Genes</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">14</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">15</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="quality-control-and-filtering" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> Quality control and filtering<a class="anchor" aria-label="anchor" href="#quality-control-and-filtering"><i class="fas fa-link"></i></a>
</h1>
<div class="inline-figure"><img src="images/workflows/02_qc.png" style="width:100.0%"></div>
<div id="learning-objectives-2" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives-2"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Recall common quality control (QC) metrics used in spatial transcriptomics, such as total transcript counts, detected genes, cell area, and negative probe percentage<br>
</li>
<li>Recall sources of technical artefacts (e.g. segmentation errors, low RNA capture, or border effects) to consider during QC<br>
</li>
<li>Describe the role of negative control probes in quantifying background noise in imaging-based technologies like CosMx<br>
</li>
<li>Generate and interpret QC visualisations (e.g. density plots, bivariate scatterplots, and FOV summaries) to identify low-quality cells<br>
</li>
<li>Evaluate how filtering thresholds affect biological interpretation, distinguishing between true low-quality cells and biologically small or lowly expressed cell types<br>
</li>
<li>Justify appropriate sample-specific filtering thresholds for metrics such as minimum transcript counts and maximum negative probe percentage</li>
</ul>
</div>
<div id="overview-2" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> Overview<a class="anchor" aria-label="anchor" href="#overview-2"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Basic QC plots + filtering (20min)</li>
<li>(Negative probes if using cosmx)</li>
<li>Identify low quality cells</li>
</ul>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span></span>
<span>  echo <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  warning <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load libraries for this step</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://here.r-lib.org/">here</a></span><span class="op">)</span></span></code></pre></div>
<p>Read in subsampled seurat object.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read in the subsampled data for this lesson</span></span>
<span><span class="va">so_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">so_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="identifying-potential-technical-artefacts" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> Identifying potential technical artefacts<a class="anchor" aria-label="anchor" href="#identifying-potential-technical-artefacts"><i class="fas fa-link"></i></a>
</h2>
<p>Library size (total transcripts) and cell areas are good indicators of technical
artefacts introduced during data generation steps.</p>
<p>Possible artefacts can include:</p>
<ul>
<li>Low quality cells<br>
</li>
<li>Cell segmentation challenges (combining two cells, not aligned with real cell
boundaries, leading to high cell area and low RNA count)</li>
</ul>
<p>Understanding how these are prevalent in your data can help with biological
interpretation downstream. For example, discerning whether differences in cells
are due to e.g. cell type differences, or technical artefacts.</p>
<p>We will first view the library size and cell area distributions on their own.</p>
<div id="library-size-transcript-counts" class="section level3" number="7.3.1">
<h3>
<span class="header-section-number">7.3.1</span> Library size (transcript counts)<a class="anchor" aria-label="anchor" href="#library-size-transcript-counts"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">nCount_RNA</span>, col <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotation_logticks.html">annotation_logticks</a></span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span>, colour <span class="op">=</span> <span class="st">"lightgrey"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Transcript counts per cell"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-23-1.png" width="672"></div>
<p>The transcripts per cell all display a log-normal distribution with shifting medians.</p>
<p>Indicates no largely low-quality cells, and a good example of library size
effects that should be normalised downstream, prior to analyses.</p>
</div>
<div id="cell-surface-area" class="section level3" number="7.3.2">
<h3>
<span class="header-section-number">7.3.2</span> Cell surface area<a class="anchor" aria-label="anchor" href="#cell-surface-area"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Total counts per cell</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Area</span>, col<span class="op">=</span><span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotation_logticks.html">annotation_logticks</a></span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Area (px^2) per cell"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-24-1.png" width="672"></div>
<p>Areas also follow log-normal distributions with medians ranging between 3,000 -
4,000px^2. Sample/FOV <code>CD.c</code> has slightly smaller cell area with a median
of ~2,000px^2.</p>
</div>
</div>
<div id="bivariate-qc" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> Bivariate QC<a class="anchor" aria-label="anchor" href="#bivariate-qc"><i class="fas fa-link"></i></a>
</h2>
<p>The next plot compares total transcript counts (nCount_RNA) with the number of
detected genes (nFeature_RNA) for each sample. The strong positive relationship
indicates that most captured transcripts are successfully mapped to known genes,
reflecting good data quality and sound probe hybridisation.</p>
<p>Only a small subset of cells deviates from this trend - typically those with low total counts. This suggests low-quality cells in the tissue sample, rather than widespread technical artefacts.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">nCount_RNA</span>, y <span class="op">=</span> <span class="va">nFeature_RNA</span>, colour <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.4</span>, size <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-25-1.png" width="672"></div>
<div class="rmdimportant">
<p><strong>Accounting for variation in cell sizes</strong></p>
<p>Cautious of samples with highly variable varying sizes. Total transcript
counts and cell areas can vary substantially. Applying “standard” QC or
filtering thresholds uniformly across all cells may remove valid cells.</p>
<div class="inline-figure"><img src="images/hatton2023_fig3.jpg" style="width:90.0%"></div>
<p><em>Fig 3. from Hatton et al. (2023): Distribution of cell mass (g) across different cell types.</em></p>
</div>
<p>The following plot shows the relationship between the total transcript counts
(nCount_RNA) and cell area across samples. In all tissue samples (Seurat FOVs),
larger cells generally contain more transcripts, forming a positive log-log
correlation.</p>
<p>However, some samples (<code>HC.a</code>, <code>HC.c</code>, <code>CD.a</code>, <code>CD.c</code>) display greater
variability and a subset of small, high-transcript cells, which may indicate
segmentation artefacts or biological differences such as the presence of smaller immune cells.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Area</span>, y<span class="op">=</span><span class="va">nCount_RNA</span>, colour<span class="op">=</span><span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotation_logticks.html">annotation_logticks</a></span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"bl"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotation_logticks.html">annotation_logticks</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"lightgrey"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-26-1.png" width="672"></div>
</div>
<div id="filtering-low-quality-cells" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> Filtering low-quality cells<a class="anchor" aria-label="anchor" href="#filtering-low-quality-cells"><i class="fas fa-link"></i></a>
</h2>
<div id="negative-probes-as-an-indicator-of-noise" class="section level3" number="7.5.1">
<h3>
<span class="header-section-number">7.5.1</span> Negative probes as an indicator of noise<a class="anchor" aria-label="anchor" href="#negative-probes-as-an-indicator-of-noise"><i class="fas fa-link"></i></a>
</h3>
<p>Spatial transcriptomics assays include both RNA-specific probes, and negative
control probes (or, background probes). While RNA specific-probes hybridise to
specific targets, negative probes do not bind to anything. Their primary purpose
is to quantify and account for this technical noise and non-specific signal.</p>
<p>In this step, we will calculate the proportion of negative control probes
relative to the total counts (RNA) for each cell. Cells exhibiting a high
percentage of noise (i.e., high proportion of negative probe counts) can be
flagged and removed during the QC stage.</p>
<p>Key metrics to add to the Seurat object:</p>
<ul>
<li>
<code>so$percent_neg</code>: The percentage of negative probes per cell</li>
<li>
<code>so$mean_neg</code>: The average negative probes per cell</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">$</span><span class="va">percent_neg</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">nCount_negprobes</span> <span class="op">/</span> <span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">nCount_RNA</span> <span class="op">+</span> <span class="va">so</span><span class="op">$</span><span class="va">nCount_negprobes</span><span class="op">)</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span></span>
<span></span>
<span><span class="co"># Join layers to calculate average_neg</span></span>
<span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/SplitLayers.html">JoinLayers</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">)</span></span>
<span><span class="va">so</span><span class="op">$</span><span class="va">mean_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colMeans</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">)</span> <span class="co"># Only defined first sample</span></span></code></pre></div>
<p>The bar plot shows that some samples (CD_b, CD_c, and HC_b) have higher
average negative probe counts, suggesting variable levels of noise across
slides.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">tissue_sample</span>, y <span class="op">=</span> <span class="va">mean_neg</span>, colour <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-28-1.png" width="672"></div>
<p>However, displaying summary metrics alone, such as the average may not
capture the variation in the data, or may be skewed. The density plot illustrates that most cells
across all samples have low levels of background signal, as indicated by the small proportion of negative
probes (medians across all samples &lt; 2%). While overall data quality
appears high, minor shifts suggest minimal between-sample variation.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">percent_neg</span>, colour <span class="op">=</span> <span class="va">tissue_sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotation_logticks.html">annotation_logticks</a></span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span>, colour <span class="op">=</span> <span class="st">"lightgrey"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-29-1.png" width="672"></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Discuss difference with Xenium - xenium lower counts, lower background.</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">mean_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-30-1.png" width="672"></div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-30-2.png" width="672"></div>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_ncount_rna</span> <span class="op">&lt;-</span> <span class="fl">90</span></span>
<span><span class="va">max_percent_neg</span> <span class="op">&lt;-</span> <span class="fl">2</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">max_percent_neg</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">min_ncount_rna</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">21</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-32-1.png" width="672"></div>
</div>
<div id="identifying-filtering-thresholds" class="section level3" number="7.5.2">
<h3>
<span class="header-section-number">7.5.2</span> Identifying filtering thresholds<a class="anchor" aria-label="anchor" href="#identifying-filtering-thresholds"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply a filter</span></span>
<span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    qc <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html">if_else</a></span><span class="op">(</span><span class="va">nCount_RNA</span> <span class="op">&gt;=</span> <span class="va">min_ncount_rna</span> <span class="op">&amp;</span> <span class="va">percent_neg</span> <span class="op">&lt;=</span> <span class="va">max_percent_neg</span>, <span class="st">"Keep"</span>, <span class="st">"Remove"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">qc</span>, <span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>##         
##          GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a
##   Keep              6811           14002            5326            4216
##   Remove            1984            1764            4393            3323
##         
##          GSM7473689.CD.b GSM7473690.CD.c
##   Keep             12479            8636
##   Remove             720            4970</code></pre>
<p>Nearly 50% of the cells for HC.c will be filtered out, display which cells will
be removed.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473684.HC.c"</span>, group.by <span class="op">=</span> <span class="st">"qc"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-34-1.png" width="672"></div>
<p>The stromal compartment will be removed entirely and can be difficult for
co-localisation analyses. Further, the epithelial compartment in the bottom left
FOV is very patchy.</p>
</div>
</div>
<div id="activity-identifying-your-own-filtering-thresholds" class="section level2 rmdnote" number="7.6">
<h2>
<span class="header-section-number">7.6</span> ACTIVITY: Identifying your own filtering thresholds<a class="anchor" aria-label="anchor" href="#activity-identifying-your-own-filtering-thresholds"><i class="fas fa-link"></i></a>
</h2>
<p>TODO: Hide this block, summarise into tips/hints</p>
<p>You have seen an example where stringent filtering omits key cells or
compartments.</p>
<p>Identify a suitable threshold for <code>min_ncount_rna</code> and <code>max_percent_neg</code> by
iteratively:</p>
<ol style="list-style-type: decimal">
<li>Changing the thresholds for <code>min_ncount_rna</code> and <code>max_percent_neg</code>
</li>
<li>Displaying the bivariate distribution of <code>nCount_RNA</code> and <code>percent_neg</code> per
tissue sample</li>
<li>Using <code>ImageDimPlot</code> to visualise which cells are removed</li>
</ol>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_ncount_rna</span> <span class="op">&lt;-</span> <span class="fl">70</span></span>
<span><span class="va">max_percent_neg</span> <span class="op">&lt;-</span> <span class="fl">4</span></span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y<span class="op">=</span><span class="va">percent_neg</span>, x<span class="op">=</span><span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>pch<span class="op">=</span><span class="fl">3</span>, alpha<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="va">max_percent_neg</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">min_ncount_rna</span>, lty<span class="op">=</span><span class="fl">3</span>, colour <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">21</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">tissue_sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Negative probes vs counts"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-36-1.png" width="672"></div>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    qc <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html">if_else</a></span><span class="op">(</span><span class="va">nCount_RNA</span> <span class="op">&gt;=</span> <span class="va">min_ncount_rna</span> <span class="op">&amp;</span> <span class="va">percent_neg</span> <span class="op">&lt;=</span> <span class="va">max_percent_neg</span>, <span class="st">"Keep"</span>, <span class="st">"Remove"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">qc</span>, <span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>##         
##          GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a
##   Keep              7411           14774            6456            5070
##   Remove            1384             992            3263            2469
##         
##          GSM7473689.CD.b GSM7473690.CD.c
##   Keep             12839           10303
##   Remove             360            3303</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Epithelial compartment less patchy, but still hard to retain the stromal cells</span></span>
<span><span class="co"># These are the trade-offs to consider</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473684.HC.c"</span>, group.by <span class="op">=</span> <span class="st">"qc"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-38-1.png" width="672"></div>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A lot of cells marked for removal and a bit patchy</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473688.CD.a"</span>, group.by <span class="op">=</span> <span class="st">"qc"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-39-1.png" width="672"></div>
</div>
<div id="applying-the-filter" class="section level3" number="7.6.1">
<h3>
<span class="header-section-number">7.6.1</span> Applying the filter<a class="anchor" aria-label="anchor" href="#applying-the-filter"><i class="fas fa-link"></i></a>
</h3>
<p>For the purposes of the workshop, we will all use the same threshold.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">min_ncount_rna</span> <span class="op">&lt;-</span> <span class="fl">70</span></span>
<span><span class="va">max_percent_neg</span> <span class="op">&lt;-</span> <span class="fl">4</span></span></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">so</span>, <span class="va">qc</span> <span class="op">==</span> <span class="st">"Keep"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so_filtered</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b 
##            7411           14774            6456            5070           12839 
## GSM7473690.CD.c 
##           10303</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">so_filtered</span>, file <span class="op">=</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_02_rna70_neg4.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="field-of-view-qc-cosmx-only-advanced" class="section level2" number="7.7">
<h2>
<span class="header-section-number">7.7</span> Field of View QC (CosMX only; Advanced)<a class="anchor" aria-label="anchor" href="#field-of-view-qc-cosmx-only-advanced"><i class="fas fa-link"></i></a>
</h2>
<p>Good to check library sizes across FOVs - red flags include FOVs with close to
no cells.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/unite.html">unite</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">unique_fov_names</span>, <span class="va">orig.ident</span>, <span class="va">fov</span>, remove <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">nCount_RNA</span>, col <span class="op">=</span> <span class="va">orig.ident</span>, group <span class="op">=</span> <span class="va">unique_fov_names</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotation_logticks.html">annotation_logticks</a></span><span class="op">(</span>sides <span class="op">=</span> <span class="st">"b"</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-43-1.png" width="672"></div>
<p>TODO: As well as border effects. Run per sample.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For each cell, calculate the distance between the centroid to each FOV border</span></span>
<span><span class="va">so_borders</span> <span class="op">&lt;-</span> <span class="va">so</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">tissue_sample</span>, <span class="va">fov</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    top <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">CenterY_local_px</span><span class="op">)</span> <span class="op">-</span> <span class="va">CenterY_local_px</span>,</span>
<span>    bottom <span class="op">=</span> <span class="va">CenterY_local_px</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">CenterY_local_px</span><span class="op">)</span>,</span>
<span>    left <span class="op">=</span> <span class="va">CenterX_local_px</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">CenterX_local_px</span><span class="op">)</span>,</span>
<span>    right <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">CenterX_local_px</span><span class="op">)</span> <span class="op">-</span> <span class="va">CenterX_local_px</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">top</span>, <span class="va">bottom</span>, <span class="va">left</span>, <span class="va">right</span>, <span class="va">nCount_RNA</span>, <span class="va">tissue_sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">nCount_RNA</span>, <span class="va">tissue_sample</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"fov_direction"</span>, values_to <span class="op">=</span> <span class="st">"distance"</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">so_borders</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">distance</span>, y <span class="op">=</span> <span class="va">nCount_RNA</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">fov_direction</span>, ncol <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_light</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">geom_smooth</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"loess"</span>, span <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="summary-1" class="section level2" number="7.8">
<h2>
<span class="header-section-number">7.8</span> Summary<a class="anchor" aria-label="anchor" href="#summary-1"><i class="fas fa-link"></i></a>
</h2>
<p>Quality control (QC) in spatial transcriptomics is a highly iterative and
context-dependent process. It requires continual exploration of your data rather
than following a prescriptive set of rules. When examining and filtering your
data for the first time, we recommend setting permissive threshold so
meaningful cells are not removed. After a preliminary look at the data, QC steps
can be revisted.</p>
<p>In comparison to older transcriptomics technologies, spatial technologies introduce
new sources of technical variation that can make interpretation of your results
tricky, and even obscure true, biological signals in the data. While you do not
need to filter in every step, it is useful to understand these potential sources
of bias, and inform subsequent rounds of QC.</p>
<p>Sources of technical variation are also platform dependent and should be
accounted for. For example, CosMx contains FOVs which adds another dimension,
compared to Xenium.</p>
<p>Quality control is not a step you conduct once during a pre-processing pipeline.
Throughout the following steps in the workshop, you will continue to conduct QC
steps.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></div>
<div class="next"><a href="normalisation-and-scaling.html"><span class="header-section-number">8</span> Normalisation and scaling</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#quality-control-and-filtering"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="nav-link" href="#learning-objectives-2"><span class="header-section-number">7.1</span> Learning objectives</a></li>
<li><a class="nav-link" href="#overview-2"><span class="header-section-number">7.2</span> Overview</a></li>
<li>
<a class="nav-link" href="#identifying-potential-technical-artefacts"><span class="header-section-number">7.3</span> Identifying potential technical artefacts</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#library-size-transcript-counts"><span class="header-section-number">7.3.1</span> Library size (transcript counts)</a></li>
<li><a class="nav-link" href="#cell-surface-area"><span class="header-section-number">7.3.2</span> Cell surface area</a></li>
</ul>
</li>
<li><a class="nav-link" href="#bivariate-qc"><span class="header-section-number">7.4</span> Bivariate QC</a></li>
<li>
<a class="nav-link" href="#filtering-low-quality-cells"><span class="header-section-number">7.5</span> Filtering low-quality cells</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#negative-probes-as-an-indicator-of-noise"><span class="header-section-number">7.5.1</span> Negative probes as an indicator of noise</a></li>
<li><a class="nav-link" href="#identifying-filtering-thresholds"><span class="header-section-number">7.5.2</span> Identifying filtering thresholds</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#activity-identifying-your-own-filtering-thresholds"><span class="header-section-number">7.6</span> ACTIVITY: Identifying your own filtering thresholds</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#applying-the-filter"><span class="header-section-number">7.6.1</span> Applying the filter</a></li></ul>
</li>
<li><a class="nav-link" href="#field-of-view-qc-cosmx-only-advanced"><span class="header-section-number">7.7</span> Field of View QC (CosMX only; Advanced)</a></li>
<li><a class="nav-link" href="#summary-1"><span class="header-section-number">7.8</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-15.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
