<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 16 Resources | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="Links to various useful resources. General guides &amp; Vignettes Single cell best practices : Comprehensive resource on single cell analysis OSCA - (Orchestrating Single Cell Analayis with...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 16 Resources | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="Links to various useful resources. General guides &amp; Vignettes Single cell best practices : Comprehensive resource on single cell analysis OSCA - (Orchestrating Single Cell Analayis with...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 16 Resources | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="Links to various useful resources. General guides &amp; Vignettes Single cell best practices : Comprehensive resource on single cell analysis OSCA - (Orchestrating Single Cell Analayis with...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="study-overview.html"><span class="header-section-number">4</span> Study overview</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="" href="normalisation-and-scaling.html"><span class="header-section-number">8</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">13</span> Spatially Restricted Genes</a></li>
<li><a class="" href="Summary.html"><span class="header-section-number">14</span> Summary</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">15</span> Next steps</a></li>
<li><a class="active" href="resources.html"><span class="header-section-number">16</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="resources" class="section level1" number="16">
<h1>
<span class="header-section-number">16</span> Resources<a class="anchor" aria-label="anchor" href="#resources"><i class="fas fa-link"></i></a>
</h1>
<p>Links to various useful resources.</p>
<p><strong>General guides &amp; Vignettes</strong></p>
<ul>
<li>
<a href="https://www.sc-best-practices.org/">Single cell best practices</a> : Comprehensive resource on single cell analysis</li>
<li>
<a href="https://bioconductor.org/books/release/OSCA/">OSCA - (Orchestrating Single Cell Analayis with bioconductor )</a> : A detailed resource on single cell analyses using the bioconductor ecosystem of packages - still applicable to certain spatial analyses.</li>
<li>
<a href="https://lmweber.org/OSTA/">OSTA (Orchestrating Spatial Transcriptomics Analysis with Bioconductor</a> : A detailed resource on how to perform spatial analyses using the bioconductor ecosystem of packages. Even if using Seurat, this is a useful resource for detailed explanations of analysis tasks.</li>
<li>
<a href="https://swbioinf.github.io/spatialsnippets/index.html">Spatial Sampler</a> : Worked examples and code snippets to run various statistical tests on single cell spatial data.</li>
<li><a href="https://www.10xgenomics.com/analysis-guides">10X analysis guides</a></li>
<li><a href="https://unclineberger.org/unclcn/wp-content/uploads/sites/871/2023/08/2023Basic-Cell-Histology-Lecture_onlinePath.pdf">Basic cell histology</a></li>
<li>
<a href="https://github.com/bhuvad/CosMxSpatialAnalysisWorkshop/tree/main">Introduction to imaging-based spatial transcriptomics analysis</a>: Workshop walking through a spatial analysis of cosmx data.</li>
</ul>
<p><strong>Analysis ecosytems </strong></p>
<ul>
<li>
<a href="https://satijalab.org/seurat/">Seurat</a>: Seurat package. Originally for single cell transcriptomics, it now supports some analyses in spatial technologies.</li>
<li>
<a href="https://www.bioconductor.org/">Bioconductor</a>: The bioconductor repository is much broader than just spatial, but various toolks centre around the <a href="https://www.bioconductor.org/packages/release/bioc/html/SpatialExperiment.html">SpatialExperiment</a> and <a href="https://bioconductor.org/packages/devel/bioc/html/SpatialFeatureExperiment.html">SpatialFeatureExperiment</a> (derived from <a href="https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html">SingleCellExperiment</a> objects). The <a href="https://lmweber.org/OSTA/">OSTA</a> book is a good overview about whats possible. With the <a href="https://pachterlab.github.io/voyager/">Voyager Documentaion</a> covering SpatialFeatureExperiment methods.</li>
<li>
<a href="https://scverse.org/">SCverse</a>: An ecosystem for analysing single cell and spatial single cell data with python.</li>
<li>
<a href="https://giottosuite.com/">Giotto Suite</a>: Another R based ecosystem for spatial analyses.</li>
<li>
<a href="https://cellprofiler.org/">Cell Profiler</a> : One toolkit often used in traditional imaging for various tasks, e.g. cell morphology quantifications.</li>
</ul>
<p><strong>Specific analyses / tools</strong></p>
<ul>
<li><p><a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2">Seurat spatial vignette (imaging-based )</a> : How to load and plot spatial data with seurat - covers different technologies.</p></li>
<li><p>[Seurat visum HD vignette][<a href="https://satijalab.org/seurat/articles/visiumhd_analysis_vignette" class="uri">https://satijalab.org/seurat/articles/visiumhd_analysis_vignette</a>] : Analysis of spot-based visium HD data.</p></li>
<li><p><a href="https://satijalab.org/seurat/articles/essential_commands.html">Seurat cheat-sheet</a> : Essential utility commands for seurat analysis, e.g merging seurat objects from different samples, subsetting, managing feilds of view, assays and layer data.</p></li>
<li><p><a href="https://satijalab.org/seurat/articles/integration_introduction">‘Introduction to scRNA-seq integration’ vignette</a>: To merge data from different experiments (including scRNAseq)</p></li>
<li><p><a href="https://satijalab.org/seurat/articles/seurat5_sketch_analysis">Seurat sketch based analysis</a> : For very large datasets.</p></li>
<li><p><a href="https://doi.org/10.1093/gigascience/giy083">Clustree</a> : A useful toolkit for choosing cluster resolutions.</p></li>
<li><p><a href="https://github.com/phipsonlab/speckle">Propeller for testing celltype proportion changes</a>: The speckle package has tools to test for changes in celltype proportions between groups</p></li>
<li><p><a href="https://bioconductor.org/packages/release/bioc/html/Statial.html">Statial package</a> and <a href="https://www.sciencedirect.com/science/article/pii/S2667237525002115">Kontextual paper</a>: Toolkit for quanitifying spatial differences between</p></li>
</ul>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="NextSteps.html"><span class="header-section-number">15</span> Next steps</a></div>
<div class="next"><a href="references.html">References</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#resources"><span class="header-section-number">16</span> Resources</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-23.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
