<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 11 Spatially Restricted Genes | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="11.1 Goals Find genes that have a particularly non-random distribution (high autocorrelation) across the tissue Investigate what might be the reason behind the high autocorrlation of one of those...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 11 Spatially Restricted Genes | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="11.1 Goals Find genes that have a particularly non-random distribution (high autocorrelation) across the tissue Investigate what might be the reason behind the high autocorrlation of one of those...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 11 Spatially Restricted Genes | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="11.1 Goals Find genes that have a particularly non-random distribution (high autocorrelation) across the tissue Investigate what might be the reason behind the high autocorrlation of one of those...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="study-background.html"><span class="header-section-number">3</span> Study background</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">4</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">5</span> Quality control and filtering</a></li>
<li><a class="" href="normalisation-and-scaling.html"><span class="header-section-number">6</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">7</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">8</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">9</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">10</span> Clustering Labeling</a></li>
<li><a class="active" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">11</span> Spatially Restricted Genes</a></li>
<li><a class="" href="Summary.html"><span class="header-section-number">12</span> Summary</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">13</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">14</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="SpatiallyRestrictedGenes" class="section level1" number="11">
<h1>
<span class="header-section-number">11</span> Spatially Restricted Genes<a class="anchor" aria-label="anchor" href="#SpatiallyRestrictedGenes"><i class="fas fa-link"></i></a>
</h1>
<div id="goals-6" class="section level2" number="11.1">
<h2>
<span class="header-section-number">11.1</span> Goals<a class="anchor" aria-label="anchor" href="#goals-6"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Find genes that have a particularly non-random distribution (high autocorrelation) across the tissue</li>
<li>Investigate what might be the reason behind the high autocorrlation of one of those genes.</li>
</ul>
</div>
<div id="overview-8" class="section level2" number="11.2">
<h2>
<span class="header-section-number">11.2</span> Overview<a class="anchor" aria-label="anchor" href="#overview-8"><i class="fas fa-link"></i></a>
</h2>
<div class="inline-figure"><img src="images/workflows/07_analysis.png" style="width:100.0%"></div>
<!--
Requires packages:
* Rfast2
* ape
-->
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT">DT</a></span><span class="op">)</span></span></code></pre></div>
<p>What genes have a spatially restricted expression across the tissue? These might be localised to a particular region and indicate some spatial activity.</p>
<p>We can calculate ‘spatial autocorrelation’ with the MoransI test to find them.</p>
</div>
<div id="run-moransi-once" class="section level2" number="11.3">
<h2>
<span class="header-section-number">11.3</span> Run MoransI once<a class="anchor" aria-label="anchor" href="#run-moransi-once"><i class="fas fa-link"></i></a>
</h2>
<p>To run this quickly, going to arbitrarily reduce the number of variable features to just 15. In real life, we might pick all our VariableFeatures.</p>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">so</span>, nfeatures<span class="op">=</span><span class="fl">15</span><span class="op">)</span></span></code></pre></div>
<p>First lets run it for a single sample by subsetting the object.</p>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">the_sample</span> <span class="op">&lt;-</span> <span class="st">'CD_a'</span></span>
<span><span class="va">the_fov</span>    <span class="op">&lt;-</span> <span class="st">'GSM7473688.CD.a'</span></span>
<span><span class="va">so_sample</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">so</span>, <span class="va">tissue_sample</span> <span class="op">==</span> <span class="va">the_sample</span><span class="op">)</span></span></code></pre></div>
<p>There’s a current bug (<a href="https://github.com/satijalab/seurat/issues/10099">see ticket</a>) with the FindSpatiallyVariableFeatures function with this kind of data. We should be able to run it directly on the seurat object itself, but as a workaround, we will run it on the ‘RNA’ assay.</p>
<div class="sourceCode" id="cb146"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If the Seurat method above crashes (segfault), a workaround is</span></span>
<span><span class="co"># running FindSpatiallyVariableFeatures at the assay level</span></span>
<span><span class="va">so_sample_assay</span> <span class="op">&lt;-</span> <span class="va">so_sample</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">tc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/GetTissueCoordinates.html">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">so_sample</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">tc</span><span class="op">)</span><span class="op">&lt;-</span> <span class="va">tc</span><span class="op">$</span><span class="va">cell</span> <span class="co"># the innards of findSpatiallyVariableFeatures expect rownames, but they're not there. Add them.</span></span>
<span></span>
<span><span class="co"># When given an 'assay' FindSpatiallyVariableFeatures returns an assay. Put it back in the seurat object</span></span>
<span><span class="va">so_sample</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindSpatiallyVariableFeatures.html">FindSpatiallyVariableFeatures</a></span><span class="op">(</span></span>
<span>  <span class="va">so_sample_assay</span>,</span>
<span>  layer <span class="op">=</span> <span class="st">"scale.data"</span>,</span>
<span>  spatial.location <span class="op">=</span> <span class="va">tc</span>, <span class="co"># now with rownnames</span></span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span><span class="va">so_sample</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span>, <span class="co"># Just a few for testing</span></span>
<span>  selection.method <span class="op">=</span> <span class="st">"moransi"</span>,</span>
<span>  nfeatures<span class="op">=</span><span class="fl">10</span> <span class="co"># mark top 10 spatially variable</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The results end up in the gene level metadata. Pull them out, but only display the genes we ran.</p>
<div class="sourceCode" id="cb147"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Put feature name as a column in the feature metadata</span></span>
<span><span class="va">so_sample</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">feature</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">so_sample</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">gene_metadata</span> <span class="op">&lt;-</span> <span class="va">so_sample</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.data</span></span>
<span></span>
<span><span class="va">gene_metadata_morans</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">gene_metadata</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">moransi.spatially.variable.rank</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">feature</span>, </span>
<span>         <span class="va">MoransI_observed</span>, <span class="va">MoransI_p.value</span>, <span class="va">moransi.spatially.variable</span>,<span class="va">moransi.spatially.variable.rank</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">moransi.spatially.variable.rank</span><span class="op">)</span></span>
<span>  </span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">gene_metadata_morans</span>, width <span class="op">=</span> <span class="st">'100%'</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-697d58abce2053a24549" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-697d58abce2053a24549">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["OLFM4","JCHAIN","KRT20","IGHM","ITLN1","CEACAM1","AQP8","CHGA","SST","TPSAB1","GCG","TPSB2","CCL21","BEST4","CCL19"],[0.30619333935982,0.2944689935180448,0.2236138360026649,0.2036567280691174,0.1196093440765579,0.09063461059843848,0.05505019557288474,0.03872614871488657,0.01785281530558092,0.01222405268991916,0.01192338776046526,0.01164827439625598,0.01103139425239425,0.007166674095836991,0.001807613828264338],[0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.000975609756097561,0.01658536585365854,0.5258536585365854],[true,true,true,true,true,true,true,true,true,true,false,false,false,false,false],[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>feature<\/th>\n      <th>MoransI_observed<\/th>\n      <th>MoransI_p.value<\/th>\n      <th>moransi.spatially.variable<\/th>\n      <th>moransi.spatially.variable.rank<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"feature","targets":1},{"name":"MoransI_observed","targets":2},{"name":"MoransI_p.value","targets":3},{"name":"moransi.spatially.variable","targets":4},{"name":"moransi.spatially.variable.rank","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script><p>What do some of these genes look like on the tissue? How can we interpret this?</p>
<div class="sourceCode" id="cb148"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so_sample</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'IGHA1'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>,</span>
<span>             border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="va">the_sample</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-114-1.png" width="672"></div>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so_sample</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'AQP8'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>,</span>
<span>             border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="va">the_sample</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-114-2.png" width="672"></div>
<div class="sourceCode" id="cb150"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so_sample</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'KRT20'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>,</span>
<span>             border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="va">the_sample</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-114-3.png" width="672"></div>
</div>
<div id="run-moransi-on-all-samples" class="section level2" number="11.4">
<h2>
<span class="header-section-number">11.4</span> Run MoransI on all samples<a class="anchor" aria-label="anchor" href="#run-moransi-on-all-samples"><i class="fas fa-link"></i></a>
</h2>
<p>Could it be that the expression pattern of a gene is disrupted? Is there a difference between samples or between groups?</p>
<p>Lets run the test on every sample. This is slow - do not run on the workshop machines, instead load the precomputed results.</p>
<p>For real life, it could be parallelised and/or run overnight.</p>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Do Not Run ##</span></span>
<span></span>
<span><span class="co"># Record moransI results for each sample, one by one.</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">tissue_sample</span><span class="op">)</span></span>
<span><span class="va">results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">the_sample</span> <span class="kw">in</span> <span class="va">samples</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">so.sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span> <span class="va">so</span>, subset<span class="op">=</span> <span class="va">tissue_sample</span> <span class="op">==</span> <span class="va">the_sample</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">so.sample.assay</span> <span class="op">&lt;-</span> <span class="va">so.sample</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">tc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/GetTissueCoordinates.html">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">so.sample</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">tc</span><span class="op">)</span><span class="op">&lt;-</span> <span class="va">tc</span><span class="op">$</span><span class="va">cell</span></span>
<span>  <span class="co"># When given an 'assay' FindSpatiallyVariableFeatures returns an assay. Put it back in the seurat object.</span></span>
<span>  <span class="va">so.sample</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindSpatiallyVariableFeatures.html">FindSpatiallyVariableFeatures</a></span><span class="op">(</span></span>
<span>    <span class="va">so.sample.assay</span>,</span>
<span>    layer <span class="op">=</span> <span class="st">"scale.data"</span>,</span>
<span>    spatial.location <span class="op">=</span> <span class="va">tc</span>,</span>
<span>    features <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/VariableFeatures.html">VariableFeatures</a></span><span class="op">(</span><span class="va">so.sample</span><span class="op">)</span>,</span>
<span>    selection.method <span class="op">=</span> <span class="st">"moransi"</span>,</span>
<span>    nfeatures<span class="op">=</span><span class="fl">15</span> <span class="co"># mark top 10 spatially variable</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="co"># --- end workaround</span></span>
<span>  </span>
<span>  </span>
<span>  <span class="co"># Format output table</span></span>
<span>  <span class="va">so.sample</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">feature</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">so.sample</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="va">gene_metadata</span> <span class="op">&lt;-</span> <span class="va">so.sample</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.data</span></span>
<span>  <span class="va">results</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gene_metadata</span>, </span>
<span>         <span class="va">feature</span>, </span>
<span>         <span class="va">MoransI_observed</span>, </span>
<span>         <span class="va">MoransI_p.value</span>, </span>
<span>         <span class="va">moransi.spatially.variable</span>,</span>
<span>         <span class="va">moransi.spatially.variable.rank</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">moransi.spatially.variable.rank</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="co"># only tested</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">moransi.spatially.variable.rank</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">the_sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">sample</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html">everything</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  </span>
<span>  <span class="va">results_list</span><span class="op">[[</span><span class="va">the_sample</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">results</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Collect output result </span></span>
<span><span class="va">results_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html">bind_rows</a></span><span class="op">(</span><span class="va">results_list</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Add group by deleting pulling out the sample name with everything after _ deleted.</span></span>
<span><span class="va">results_all</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_.*"</span>,<span class="st">""</span>, <span class="va">results_all</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span>
<span></span>
<span><span class="va">morans_results_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"morans_results.RDS"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">results_all</span>, <span class="va">morans_results_file</span><span class="op">)</span></span></code></pre></div>
<p>Load preprepared results file</p>
<div class="sourceCode" id="cb152"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">morans_results_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"morans_results.RDS"</span><span class="op">)</span></span>
<span><span class="va">results_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">morans_results_file</span><span class="op">)</span></span></code></pre></div>
<p>Lets visualise our MoransI scores for each gene for each sample on a boxplot. Note that this isn’t an established standard way of interpreting these scores, but it does give us a way to prioritise what to look at.</p>
<p>MoransI scores are between -1 (anticorrelated) and 1 (perfectly correlated) - 0 is random. The baseline isn’t 0 - probably because the observations are already limited to the tissue. But there’s still clearly some genes with higher scores than others, even though the variation between samples is quite high.</p>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># unique set of genes that were top 10 morans in any sample</span></span>
<span><span class="va">plot_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">results_all</span>, <span class="va">moransi.spatially.variable.rank</span> <span class="op">&lt;=</span> <span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">feature</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">results_all</span>, <span class="va">feature</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">plot_genes</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">feature</span>, y<span class="op">=</span><span class="va">MoransI_observed</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span> mapping<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>col<span class="op">=</span><span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"MoransI Per sample"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">group</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-117-1.png" width="960"></div>
</div>
<div id="explore-moransi-results" class="section level2" number="11.5">
<h2>
<span class="header-section-number">11.5</span> Explore moransI results<a class="anchor" aria-label="anchor" href="#explore-moransi-results"><i class="fas fa-link"></i></a>
</h2>
<p>That doesn’t mean much in isolation, so lets look at the actual distribution of some of those.</p>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so.CD_a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span> <span class="va">so</span>, subset<span class="op">=</span> <span class="va">tissue_sample</span> <span class="op">==</span> <span class="st">'CD_a'</span><span class="op">)</span></span>
<span><span class="va">so.HC_a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span> <span class="va">so</span>, subset<span class="op">=</span> <span class="va">tissue_sample</span> <span class="op">==</span> <span class="st">'HC_a'</span><span class="op">)</span></span></code></pre></div>
<p>IGHM has high autocorrelation all over</p>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.CD_a</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'IGHM'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>, border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.HC_a</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'IGHM'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>, border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> </span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-119-1.png" width="672"></div>
<p>Contrast that with a CHGA - still a highly variable gene, with a low(er) spatial autocorrelation</p>
<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.CD_a</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'CHGA'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>, border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.HC_a</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'CHGA'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>, border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> </span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-120-1.png" width="672"></div>
<p>CCL21 has a combination of high and low scores. Why?</p>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.CD_a</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'CCL21'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>, border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.HC_a</span>, </span>
<span>             molecules <span class="op">=</span> <span class="st">'CCL21'</span>, </span>
<span>             group.by <span class="op">=</span> <span class="st">'tissue_sample'</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey30"</span><span class="op">)</span>, <span class="co"># Make all cells grey.</span></span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>, border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span> </span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-121-1.png" width="672"></div>
<p>There’s a big clump of CCL21 at the bottom of HC_a, lets figure out which FOV that is and zoom in further. Now we can practically bring back the cluster labels.</p>
<div class="sourceCode" id="cb158"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.HC_a</span>, group.by <span class="op">=</span> <span class="st">'fov'</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-122-1.png" width="672"></div>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so.HC_a.4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">so.HC_a</span>, <span class="va">fov</span><span class="op">==</span><span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.HC_a.4</span>, </span>
<span>             molecules  <span class="op">=</span> <span class="st">'CCL21'</span>, </span>
<span>             group.by   <span class="op">=</span> <span class="st">'cluster_labels'</span>, alpha<span class="op">=</span><span class="fl">0.3</span>,<span class="co"># Show clusters, but faded </span></span>
<span>             cols <span class="op">=</span> <span class="st">'polychrome'</span>,</span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>,  border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-122-2.png" width="672"></div>
<p>That seems to be because of a group of T cells expressing CCL21’s. We can highlight those alongside our transcripts.</p>
<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Show only B and T cells</span></span>
<span><span class="va">so.HC_a.4</span><span class="op">$</span><span class="va">isBT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">so.HC_a.4</span><span class="op">$</span><span class="va">cluster_labels</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"c5: T cell"</span>, <span class="st">"c8: B cell"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">so.HC_a.4</span><span class="op">$</span><span class="va">cluster_labels</span><span class="op">)</span>, <span class="st">"Other"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so.HC_a.4</span>, </span>
<span>             molecules  <span class="op">=</span> <span class="st">'CCL21'</span>, </span>
<span>             group.by   <span class="op">=</span> <span class="st">'isBT'</span>, alpha<span class="op">=</span><span class="fl">0.3</span>,<span class="co"># Show clusters, but faded </span></span>
<span>             cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"c5: T cell"</span><span class="op">=</span><span class="st">'royalblue'</span>, <span class="st">"c8: B cell"</span><span class="op">=</span><span class="st">'aquamarine'</span>, <span class="st">'Other'</span><span class="op">=</span><span class="st">'grey80'</span><span class="op">)</span>,</span>
<span>             boundaries <span class="op">=</span> <span class="st">"segmentation"</span>,  border.color <span class="op">=</span> <span class="cn">NA</span>, axes <span class="op">=</span> <span class="cn">T</span>, crop<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-123-1.png" width="672"></div>
<p>If we look at Garrido-Trigo et al’s paper - they highlight these kind of lymphoid niches, full of T, B and plasma cells. They go on to look at macrophage subtypes within these regions across different conditions.</p>
<div class="inline-figure"><img src="images/GTFig2g_niche.png" style="width:100.0%"></div>
<p>In our case, these could be more formally defined by a niche-neighbourhood analysis; which at the most basic level, classifies each cell into a ‘niche’ depending on what type of cells are in its immediate vicinity. We could then analyse our data at the ‘niche’ level.</p>
<!-- ## Co-occurrence tests? (Potential addition) -->
<!-- Try to test foc co-occurrence of CCL21 and cluster (or MS4A1) -->
<!-- Easiest way might be a conversion to SFE then use voyager as per the bivariate stats workflow: https://pachterlab.github.io/voyager/articles/bivariate.html -->

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="ClusterLabelling.html"><span class="header-section-number">10</span> Clustering Labeling</a></div>
<div class="next"><a href="Summary.html"><span class="header-section-number">12</span> Summary</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#SpatiallyRestrictedGenes"><span class="header-section-number">11</span> Spatially Restricted Genes</a></li>
<li><a class="nav-link" href="#goals-6"><span class="header-section-number">11.1</span> Goals</a></li>
<li><a class="nav-link" href="#overview-8"><span class="header-section-number">11.2</span> Overview</a></li>
<li><a class="nav-link" href="#run-moransi-once"><span class="header-section-number">11.3</span> Run MoransI once</a></li>
<li><a class="nav-link" href="#run-moransi-on-all-samples"><span class="header-section-number">11.4</span> Run MoransI on all samples</a></li>
<li><a class="nav-link" href="#explore-moransi-results"><span class="header-section-number">11.5</span> Explore moransI results</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-27.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
