<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 Study background | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="3.1 Goals Understand the experimental setup and samples used for the workshop Note the differences between CosMx and Xenium data during imaging  3.2 Background This workshop will use CosMx Spatial...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 3 Study background | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="3.1 Goals Understand the experimental setup and samples used for the workshop Note the differences between CosMx and Xenium data during imaging  3.2 Background This workshop will use CosMx Spatial...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 3 Study background | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="3.1 Goals Understand the experimental setup and samples used for the workshop Note the differences between CosMx and Xenium data during imaging  3.2 Background This workshop will use CosMx Spatial...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="active" href="study-background.html"><span class="header-section-number">3</span> Study background</a></li>
<li><a class="" href="understanding-our-spatial-data.html"><span class="header-section-number">4</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">5</span> Quality control and filtering</a></li>
<li><a class="" href="normalisation-and-scaling.html"><span class="header-section-number">6</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">7</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">8</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">9</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">10</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">11</span> Spatially Restricted Genes</a></li>
<li><a class="" href="Summary.html"><span class="header-section-number">12</span> Summary</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">13</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">14</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="study-background" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Study background<a class="anchor" aria-label="anchor" href="#study-background"><i class="fas fa-link"></i></a>
</h1>
<div id="goals" class="section level2" number="3.1">
<h2>
<span class="header-section-number">3.1</span> Goals<a class="anchor" aria-label="anchor" href="#goals"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Understand the experimental setup and samples used for the workshop</li>
<li>Note the differences between CosMx and Xenium data during imaging</li>
</ul>
</div>
<div id="background" class="section level2" number="3.2">
<h2>
<span class="header-section-number">3.2</span> Background<a class="anchor" aria-label="anchor" href="#background"><i class="fas fa-link"></i></a>
</h2>
<p>This workshop will use CosMx Spatial Molecular Imager (SMI) data from the paper
<em>Macrophage and neutrophil heterogeneity
at single-cell spatial resolution in human inflammatory bowel disease</em>
(Garrido-Trigo et al. 2023)<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://satijalab.org/seurat/articles/seurat5_sketch_analysis" class="uri"&gt;https://satijalab.org/seurat/articles/seurat5_sketch_analysis&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>.</p>
<p>This data set was chosen because the authors have made their raw and
annotated data available, along with the corresponding analysis
<a href="https://github.com/HelenaLC/CosMx-SMI-IBD">code</a> and a browsable
<a href="https://servidor2-ciberehd.upc.es/external/garrido/app/">data interface</a>.</p>
<p>The study provides a clear biological contrast between health and disease
states. This disease cohort consists of donors with active <strong>inflammatory bowel
disease (IBD)</strong>, a chronic inflammatory condition of the gastrointestinal (GI) tract,
where <strong>immune cell organisation within the tissue plays a central role in
disease pathogenesis.</strong></p>
<p><strong>Inflammatory bowel disease (IBD) is an umbrella term</strong> that includes two
main GI tract disorders with different characteristics:</p>
<ul>
<li><strong>Crohn’s disease (CD)</strong></li>
<li>
<strong>ulcerative colitis (UC)</strong>.</li>
</ul>
<p>Spatial transcriptomic technologies enable gene expression to be measured at
subcellular resolution while preserving the tissue architecture, allowing
researchers to uncover spatial patterns and cell-cell interactions that would be
lost in conventional bulk or single-cell RNA-seq data.</p>
</div>
<div id="the-experimental-design-and-cohorts" class="section level2" number="3.3">
<h2>
<span class="header-section-number">3.3</span> The experimental design and cohorts<a class="anchor" aria-label="anchor" href="#the-experimental-design-and-cohorts"><i class="fas fa-link"></i></a>
</h2>
<p><img src="images/garrido-trigo_fig1a.png" style="width:100.0%"><em>Fig 1a from Garrido-Trigo et al. (2023).</em></p>
<p>The study included 9 CosMx slides of colonic biopsies, from a total of 9 donors:</p>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th>Cohorts</th>
<th>Abbreviated</th>
<th>Number of samples</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Healthy controls</strong></td>
<td>HC</td>
<td>3</td>
</tr>
<tr class="even">
<td>Ulcerative colitis</td>
<td>UC</td>
<td>3</td>
</tr>
<tr class="odd">
<td><strong>Crohn’s disease</strong></td>
<td>CD</td>
<td>3</td>
</tr>
</tbody>
</table></div>
<p>The study includes <strong>three biological replicates per condition</strong> (HC, UC, CD).
This <strong>provides enough variation to capture differences between individuals</strong> while
maintaining manageable data size for hands-on analysis.</p>
<p>In this workshop, <strong>we will focus on the healthy control (HC)
and Crohn’s disease (CD) samples.</strong></p>
<p>In the original paper, the CosMX SMI cells were annotated via label transfer from
the scRNA data. <strong>The spatial transcriptomics data will be processed independent
of the scRNA data to demonstrate how differences in cellular composition and
spatial organisation can be inferred directly from the spatial CosMx data</strong>.</p>
</div>
<div id="data-generation-slides-samples-and-field-of-views" class="section level2" number="3.4">
<h2>
<span class="header-section-number">3.4</span> Data generation: slides, samples, and field of views<a class="anchor" aria-label="anchor" href="#data-generation-slides-samples-and-field-of-views"><i class="fas fa-link"></i></a>
</h2>
<p>This section will provide an overview of the high-level data generation steps
that are conducted prior to any preprocessing or analysis steps. The steps begin
from sample preparation, to sequencing of field of views (FOVs) and cell
segmentation.</p>
<div class="inline-figure"><img src="images/30_data/data_generation_slim.png" style="width:100.0%"></div>
<div id="sample-preparation" class="section level3" number="3.4.1">
<h3>
<span class="header-section-number">3.4.1</span> Sample preparation<a class="anchor" aria-label="anchor" href="#sample-preparation"><i class="fas fa-link"></i></a>
</h3>
<p>In imaging-based spatial transcriptomics (im-SRT) workflows, tissue samples
are prepared on slides. You may have a single, or multiple tissue samples per
slide. This study utilises both for the colonic tissue.</p>
<div class="inline-figure"><img src="images/30_data/slides.png" style="width:100.0%"></div>
</div>
<div id="sequencing" class="section level3" number="3.4.2">
<h3>
<span class="header-section-number">3.4.2</span> Sequencing<a class="anchor" aria-label="anchor" href="#sequencing"><i class="fas fa-link"></i></a>
</h3>
<p>In CosMx workflows, fields of views (FOVs) must be selected from the prepared
slides. These are the spatial areas where sequencing will be conducted.</p>
<p>This step allows you to be selective of the regions, such as those of most
importance to the research question or key hypotheses, and avoids sequencing
areas that do not contain any tissue. For example, in highly heterogeneous
tumour microenvironments, you can focus on selecting boundaries where
infiltrated immune cells interact with cancerous cells.</p>
<div class="rmdimportant">
<p><strong>Xenium does not have FOVs</strong></p>
<p>Xenium uses a wide-field fluorescence microscope for automated image
acquisition across a larger, continuous imaging area. This approach allows
Xenium to map gene expression over the entire tissue region selected by the
user, rather than stitching together multiple small, distinct FOV images.</p>
</div>
<p>For the workshop, you will be using a subset of FOVs - the first four FOVs from
each of the six samples.</p>
<div class="inline-figure"><img src="images/30_data/fov_subsample.png" style="width:100.0%"></div>
</div>
<div id="cell-segmentation" class="section level3" number="3.4.3">
<h3>
<span class="header-section-number">3.4.3</span> Cell segmentation<a class="anchor" aria-label="anchor" href="#cell-segmentation"><i class="fas fa-link"></i></a>
</h3>
<p>Cell segmentation is the computational step in imaging-based ST that
identifies the boundaries of individual cells so that RNA molecules can be
correctly assigned to each one. These utilise the stained images to identify
cell border shapes that are necessary for cell-level analyses downstream.</p>
<p>The following images are of <strong>Xenium</strong> using two different staining approaches,
H&amp;E and immunofluorescent respectively,
with the detected cell boundaries overlayed on both.</p>
<p>The H&amp;E stain is often conducted as part of im-SRT protocols and is used to determine basic cell lineages from the morphology. In CosMx, this is often used to preview the slide for FOV selection.</p>
<div class="inline-figure"><img src="images/30_data/xenium_explorer_he.png" style="width:100.0%"></div>
<p>The immunoflourescent stain demarcates more specific subcellular components. This stain is often used in the automated on-machine cell segmentation step.</p>
<div class="inline-figure"><img src="images/30_data/xenium_explorer_dapi.png" style="width:100.0%"></div>
<p>Accurate segmentation is essential because it determines how reliably
transcripts are assigned to individual cells, which directly affects
downstream analyses. Commercial platforms like CosMx and Xenium include on-machine
segmentation.</p>
<p>However, you may choose to re-run segmentation using independent tools such as
Cellpose (image-based) or Baysor (transcript-based or hybrid) to improve
boundary accuracy and adapt to specific tissue types.</p>
<div class="rmdwarning">
<p><strong>H&amp;E data is not available for this study</strong></p>
<p>Spatial data analysis steps utilise H&amp;E images for quality control steps and
visualisation. For the workshop, the transcriptomics and spatial data will be
used to inform these steps.</p>
</div>

</div>
</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="intro.html"><span class="header-section-number">2</span> Introduction</a></div>
<div class="next"><a href="understanding-our-spatial-data.html"><span class="header-section-number">4</span> Understanding our spatial data</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#study-background"><span class="header-section-number">3</span> Study background</a></li>
<li><a class="nav-link" href="#goals"><span class="header-section-number">3.1</span> Goals</a></li>
<li><a class="nav-link" href="#background"><span class="header-section-number">3.2</span> Background</a></li>
<li><a class="nav-link" href="#the-experimental-design-and-cohorts"><span class="header-section-number">3.3</span> The experimental design and cohorts</a></li>
<li>
<a class="nav-link" href="#data-generation-slides-samples-and-field-of-views"><span class="header-section-number">3.4</span> Data generation: slides, samples, and field of views</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#sample-preparation"><span class="header-section-number">3.4.1</span> Sample preparation</a></li>
<li><a class="nav-link" href="#sequencing"><span class="header-section-number">3.4.2</span> Sequencing</a></li>
<li><a class="nav-link" href="#cell-segmentation"><span class="header-section-number">3.4.3</span> Cell segmentation</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-27.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
