<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 Understanding our spatial data | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="6.1 Overview Explain e.g. cosmx in detail, refer to similarities in e.g. xenium Load data into object Brief explanation of different data structures Sample annotation: Applying sample names to...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 6 Understanding our spatial data | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="6.1 Overview Explain e.g. cosmx in detail, refer to similarities in e.g. xenium Load data into object Brief explanation of different data structures Sample annotation: Applying sample names to...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 Understanding our spatial data | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="6.1 Overview Explain e.g. cosmx in detail, refer to similarities in e.g. xenium Load data into object Brief explanation of different data structures Sample annotation: Applying sample names to...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="exampleexperiment.html"><span class="header-section-number">4</span> Example Experiment</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="active" href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="" href="basic-visualisation.html"><span class="header-section-number">8</span> Basic visualisation</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">13</span> Spatially Restricted Genes</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">14</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">15</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="understanding-our-spatial-data" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> Understanding our spatial data<a class="anchor" aria-label="anchor" href="#understanding-our-spatial-data"><i class="fas fa-link"></i></a>
</h1>
<div id="overview-1" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> Overview<a class="anchor" aria-label="anchor" href="#overview-1"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Explain e.g. cosmx in detail, refer to similarities in e.g. xenium</li>
<li>Load data into object</li>
<li>Brief explanation of different data structures</li>
<li>Sample annotation: Applying sample names to individual slides. Mention difficulty of multiple samples per slide (can we recommend some approach?)</li>
</ul>
<p>What data do you get? How do these differ between CosMX and Xenium?</p>
<p>High-resolution molecular data:</p>
<ul>
<li>Expression matrix</li>
<li>Transcript coordinates</li>
<li>Read quality</li>
</ul>
<p>Spatial and imaging data:</p>
<ul>
<li>Cell segmentations</li>
<li>Image files</li>
</ul>
</div>
<div id="how-the-data-was-generated" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> How the data was generated<a class="anchor" aria-label="anchor" href="#how-the-data-was-generated"><i class="fas fa-link"></i></a>
</h2>
</div>
<div id="the-raw-data" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> The raw data<a class="anchor" aria-label="anchor" href="#the-raw-data"><i class="fas fa-link"></i></a>
</h2>
<p>For a cosmx SMI slide (like this experiment).</p>
<ul>
<li>
<strong>Flat files</strong> : Most of what we need to do processing. Typically a directory of flat files is read in as a whole, and we don’t need to worry too much. But note there are no images in here.
<ul>
<li>SLIDE-polygons.csv.gz : Cell borders</li>
<li>SLIDE_exprMat_file.csv.gz : Counts of genes per cell (Counts matrix)</li>
<li>SLIDE_fov_positions_file.csv.gz : Location of FOVs on slide</li>
<li>SLIDE_metadata_file.csv.gz : Cell level QC metadata</li>
<li>SLIDE_tx_file.csv.gz : Location of individual transcripts.</li>
</ul>
</li>
<li>
<strong>Raw files</strong> : Giant ugly directory with lots of files including microscopy images.
<ul>
<li>RawFiles/SLIDE/RUN_CODE/CellStatsDir/Morphology2D : Location of images</li>
<li><a href="https://github.com/Nanostring-Biostats/CosMxDACustomModules/blob/main/Export/CosMxDAExportSetup.docx" class="uri">https://github.com/Nanostring-Biostats/CosMxDACustomModules/blob/main/Export/CosMxDAExportSetup.docx</a></li>
</ul>
</li>
</ul>
<p>For a Xenium slide there is just the one typical output directory; they discuss the formats in the section on <a href="https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-archive-data">data archiving</a>, but again, software tools take the directory as a whole.</p>
</div>
<div id="cell-segmentation" class="section level2" number="6.4">
<h2>
<span class="header-section-number">6.4</span> Cell segmentation<a class="anchor" aria-label="anchor" href="#cell-segmentation"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Out of scope for workshop</li>
<li>commercial options</li>
<li>Third party tools often recommended: image-based (cellpose), transcript-based (proseg), hybrid (baysor), segmentation-free approaches</li>
<li>Deep learning commonly applied here</li>
</ul>
</div>
<div id="note-on-seuratloadnanostring" class="section level2" number="6.5">
<h2>
<span class="header-section-number">6.5</span> Note on <code>Seurat::LoadNanostring()</code><a class="anchor" aria-label="anchor" href="#note-on-seuratloadnanostring"><i class="fas fa-link"></i></a>
</h2>
<p>Load the libraries required for this step.</p>
<pre><code>## Loading required package: SeuratObject</code></pre>
<pre><code>## Loading required package: sp</code></pre>
<pre><code>## 
## Attaching package: 'SeuratObject'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, t</code></pre>
<pre><code>## ── Attaching core tidyverse packages ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.5
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ ggplot2   3.5.2     ✔ tibble    3.3.0
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.1
## ✔ purrr     1.1.0     
## ── Conflicts ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors
## here() starts at /home/s.williams/projects/intro-spatial-transcriptomics-workshop</code></pre>
<p>To load the data, we can use the <code><a href="https://satijalab.org/seurat/reference/ReadNanostring.html">Seurat::LoadNanostring()</a></code> function. However,
this default method drops most of the metadata in the Seurat object. For example,
what field of view (FoV) each cell a member of is missing.</p>
<p>We provide an alternate function. In time, this should be fixed within Seurat.
See comments <a href="https://github.com/satijalab/seurat/discussions/9261">here</a></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"scripts/LoadNanostring_edited_function.R"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">example_path</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="op">(</span><span class="st">"raw_data"</span>, <span class="st">"GSM7473682_HC_a"</span><span class="op">)</span></span>
<span><span class="va">example_so</span> <span class="op">&lt;-</span> <span class="fu">LoadNanostring</span><span class="op">(</span><span class="va">example_path</span>, assay<span class="op">=</span><span class="st">'RNA'</span>, fov<span class="op">=</span><span class="st">"GSM7473682_HC_a"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_so</span></span></code></pre></div>
<pre class="console"><code>An object of class Seurat 
999 features across 42045 samples within 1 assay 
Active assay: RNA (999 features, 0 variable features)
 1 layer present: counts
 1 spatial field of view present: GSM7473682_HC_a</code></pre>
<p>Explain what’s missing</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#so$condition       &lt;- "Healthy Controls"</span></span>
<span><span class="co">#so$tissue_sample   &lt;- "HC_A" # denotes healthy control, from sample A</span></span></code></pre></div>
</div>
<div id="loading-all-samples-from-raw-data" class="section level2" number="6.6">
<h2>
<span class="header-section-number">6.6</span> Loading all samples from raw data<a class="anchor" aria-label="anchor" href="#loading-all-samples-from-raw-data"><i class="fas fa-link"></i></a>
</h2>
<p>We use a subsampled data for this workshop, for three healthy control samples (HC)
and three Crohn’s disease samples (CD).</p>
<p>The first four field of views are used.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># How was this created?</span></span>
<span><span class="va">seurat_file_00_raw</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw.RDS"</span><span class="op">)</span></span>
<span><span class="va">so_raw_full</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">seurat_file_00_raw</span><span class="op">)</span></span>
<span><span class="va">so_raw_subset</span> <span class="op">&lt;-</span> <span class="va">so_raw_full</span><span class="op">[</span>, <span class="va">so_raw_full</span><span class="op">$</span><span class="va">group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD"</span>, <span class="st">"HC"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">so_raw_full</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">fov</span> <span class="op">&lt;=</span> <span class="fl">4</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">so_raw_subset</span>, <span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="loading-the-workshop-data" class="section level2" number="6.7">
<h2>
<span class="header-section-number">6.7</span> Loading the workshop data<a class="anchor" aria-label="anchor" href="#loading-the-workshop-data"><i class="fas fa-link"></i></a>
</h2>
<p>TODO: Add image for all FOVs, and outline the ones we are using</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so_path</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">so_path</span><span class="op">)</span></span></code></pre></div>
<div id="understanding-the-seuratobject" class="section level3" number="6.7.1">
<h3>
<span class="header-section-number">6.7.1</span> Understanding the <code>SeuratObject</code><a class="anchor" aria-label="anchor" href="#understanding-the-seuratobject"><i class="fas fa-link"></i></a>
</h3>
<div id="why-seurat" class="section level4 callout-info" number="6.7.1.1">
<h4>
<span class="header-section-number">6.7.1.1</span> Why Seurat?<a class="anchor" aria-label="anchor" href="#why-seurat"><i class="fas fa-link"></i></a>
</h4>
<p>Seurat objects are a data structure to store the count data and additional metadata.
As the data is very high dimensional, it is important
that connected information about cells, samples, and data stay connected
during data analysis.</p>
<p>Other frameworks such as <code>scverse</code>, and data classes like <code>anndata</code> and
<code>SpatialFeatureExperiment</code> represent the data differently, but all meet the
needs above. Differences in compatibility with other tools.</p>
<p>Flexibility across single-cell transcriptomics.</p>
</div>
<p>Display the Seurat object.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span></span></code></pre></div>
<p><strong>999 features across 68624 samples with 2 assays</strong></p>
<ul>
<li>980 genes + 19 negprobes</li>
<li>68k number of segmented cells analysed</li>
<li>two different types of measurements, detailed below</li>
</ul>
<p><strong>Active assay: RNA (980 features, 0 variable features):</strong></p>
<ul>
<li>RNA assay: Primary gene expression measurement data (counts)</li>
<li>980 transcripts measured</li>
<li>no variable features as feature selection has not been conducted yet</li>
</ul>
<p><strong>6 layers present: counts.GSM7473682.HC.a, counts.GSM7473683.HC.b,
counts.GSM7473684.HC.c, counts.GSM7473688.CD.a, counts.GSM7473689.CD.b,
counts.GSM7473690.CD.c</strong></p>
<ul>
<li>Each layer represents a sample from either HC or CD donors. Required for
comparison of both groups downstream</li>
</ul>
<p><strong>1 other assay present: negprobes</strong></p>
<p>This secondary assay specifically stores the counts corresponding to negative
control probes. In spatial transcriptomics, tracking these control probes is
crucial for quality control, specificity assessment, and defining background
signal.</p>
<p><strong>6 spatial fields of view present: GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b GSM7473690.CD.c:</strong></p>
<p>The list of six spatial fields of view (FOVs) explicitly links the spatial/image
information stored in the Seurat object to the six specific samples/layers
identified above. This demonstrates that the data set is ready for spatially
resolved analysis and visualisation, allowing results (like cell clusters or
gene expression) to be mapped back onto the original tissue sections for all
six integrated samples.</p>
<div id="cosmx-sample-names" class="section level4 callout-tip" number="6.7.1.2">
<h4>
<span class="header-section-number">6.7.1.2</span> CosMX sample names<a class="anchor" aria-label="anchor" href="#cosmx-sample-names"><i class="fas fa-link"></i></a>
</h4>
<p>Note that the field of view names are delimited by <code>.</code>. At the time of writing,
underscores will cause some functions to fail silently. For more information,
see <a href="https://github.com/satijalab/seurat/issues/10103">this issue</a></p>
</div>
<div id="xenium" class="section level4 callout-tip" number="6.7.1.3">
<h4>
<span class="header-section-number">6.7.1.3</span> Xenium<a class="anchor" aria-label="anchor" href="#xenium"><i class="fas fa-link"></i></a>
</h4>
<p>Xenium does not have FoVs.</p>
<p>TODO: expand</p>
</div>
<p>Display the metadata</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<pre><code>##               orig.ident nCount_RNA nFeature_RNA fov cell_ID  Area AspectRatio
## HC_a_1_1 GSM7473682.HC.a        366          187   1       1  6153        0.67
## HC_a_2_1 GSM7473682.HC.a        805          283   1       2 16178        0.85
## HC_a_3_1 GSM7473682.HC.a        118           73   1       3  3119        1.52
## HC_a_4_1 GSM7473682.HC.a        106           59   1       4  3988        1.52
## HC_a_5_1 GSM7473682.HC.a        462          207   1       5  4773        0.53
## HC_a_6_1 GSM7473682.HC.a       1314          436   1       6 11588        0.87
##          CenterX_local_px CenterY_local_px CenterX_global_px CenterY_global_px
## HC_a_1_1             2119             3443          19274.56          173198.6
## HC_a_2_1             1847             3404          19002.56          173159.6
## HC_a_3_1             1986             3458          19141.56          173213.6
## HC_a_4_1             2362             3454          19517.56          173209.6
## HC_a_5_1             2159             3415          19314.56          173170.6
## HC_a_6_1             1664             3402          18819.56          173157.6
##          Width Height Mean.MembraneStain Max.MembraneStain Mean.PanCK Max.PanCK
## HC_a_1_1    79    118                680              2218      17170     29735
## HC_a_2_1   137    161                615              4453      16775     29702
## HC_a_3_1    79     52                989              2827      24033     29662
## HC_a_4_1    91     60                449              2099      13575     29651
## HC_a_5_1    63    118               1002              2425      24946     29670
## HC_a_6_1   116    134               1024              2320      25700     29712
##          Mean.CD45 Max.CD45 Mean.CD3 Max.CD3 Mean.DAPI Max.DAPI cell
## HC_a_1_1       198     9639      234    1237        20      131  1_1
## HC_a_2_1       222    21371      234    1520        12      122  2_1
## HC_a_3_1       171     6858      298    1721        13       58  3_1
## HC_a_4_1       146      935      206    1402         6       56  4_1
## HC_a_5_1       299     6387      330    1443        22      116  5_1
## HC_a_6_1       354    23243      328    1610        23      113  6_1
##          individual_code tissue_sample group        condition fov_name
## HC_a_1_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_2_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_3_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_4_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_5_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_6_1            HC_a          HC_a    HC Healthy controls HC_a_001
##          nCount_negprobes nFeature_negprobes
## HC_a_1_1                2                  2
## HC_a_2_1                5                  3
## HC_a_3_1                1                  1
## HC_a_4_1                0                  0
## HC_a_5_1                3                  2
## HC_a_6_1                8                  5</code></pre>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th>Column</th>
<th>Definition</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Row name</td>
<td>The sample and cell</td>
</tr>
<tr class="even">
<td><code>orig.ident</code></td>
<td>Sample</td>
</tr>
<tr class="odd">
<td><code>nCount RNA</code></td>
<td>The number of sequenced transcripts</td>
</tr>
<tr class="even">
<td><code>nFeature_RNA</code></td>
<td>The number of transcripts mapped to a gene</td>
</tr>
</tbody>
</table></div>
<p>Each sample should be annotated with its experimental details. This is important
so we know which groups to compare downstream, i.e. case vs. control comparison.
This particular study has one sample per slide (easy!), but there are typically
more.</p>
<p>TODO: Explain input data, what it represents, how it will be used for analysis.</p>
<p>TODO: Define terms like centroids, segments, fov</p>
<ul>
<li>See: <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2#mouse-brain-vizgen-merscope" class="uri">https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2#mouse-brain-vizgen-merscope</a>
</li>
<li>TODO: Excalidraw of these</li>
</ul>
<p>Display the metadata columns:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<pre><code>##  [1] "orig.ident"         "nCount_RNA"         "nFeature_RNA"      
##  [4] "fov"                "cell_ID"            "Area"              
##  [7] "AspectRatio"        "CenterX_local_px"   "CenterY_local_px"  
## [10] "CenterX_global_px"  "CenterY_global_px"  "Width"             
## [13] "Height"             "Mean.MembraneStain" "Max.MembraneStain" 
## [16] "Mean.PanCK"         "Max.PanCK"          "Mean.CD45"         
## [19] "Max.CD45"           "Mean.CD3"           "Max.CD3"           
## [22] "Mean.DAPI"          "Max.DAPI"           "cell"              
## [25] "individual_code"    "tissue_sample"      "group"             
## [28] "condition"          "fov_name"           "nCount_negprobes"  
## [31] "nFeature_negprobes"</code></pre>
<p>There is a lot of information. Use something like skimr for EDA and to
understand each field at a glance</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">skimr</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/skimr/reference/skim.html">skim</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:unnamed-chunk-13">Table 6.1: </span>Data summary</caption>
<tbody>
<tr class="odd">
<td align="left">Name</td>
<td align="left"><a href="mailto:so@meta.data" class="email">so@meta.data</a></td>
</tr>
<tr class="even">
<td align="left">Number of rows</td>
<td align="left">68624</td>
</tr>
<tr class="odd">
<td align="left">Number of columns</td>
<td align="left">31</td>
</tr>
<tr class="even">
<td align="left">_______________________</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Column type frequency:</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">character</td>
<td align="left">7</td>
</tr>
<tr class="odd">
<td align="left">numeric</td>
<td align="left">24</td>
</tr>
<tr class="even">
<td align="left">________________________</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Group variables</td>
<td align="left">None</td>
</tr>
</tbody>
</table></div>
<p><strong>Variable type: character</strong></p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="21%">
<col width="13%">
<col width="18%">
<col width="5%">
<col width="5%">
<col width="8%">
<col width="12%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">skim_variable</th>
<th align="right">n_missing</th>
<th align="right">complete_rate</th>
<th align="right">min</th>
<th align="right">max</th>
<th align="right">empty</th>
<th align="right">n_unique</th>
<th align="right">whitespace</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">orig.ident</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">cell</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">18518</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">individual_code</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">tissue_sample</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">group</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">condition</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">16</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">fov_name</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">0</td>
</tr>
</tbody>
</table></div>
<p><strong>Variable type: numeric</strong></p>
<div class="inline-table"><table style="width:100%;" class="table table-sm">
<colgroup>
<col width="16%">
<col width="8%">
<col width="11%">
<col width="8%">
<col width="7%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">skim_variable</th>
<th align="right">n_missing</th>
<th align="right">complete_rate</th>
<th align="right">mean</th>
<th align="right">sd</th>
<th align="right">p0</th>
<th align="right">p25</th>
<th align="right">p50</th>
<th align="right">p75</th>
<th align="right">p100</th>
<th align="left">hist</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">nCount_RNA</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">224.95</td>
<td align="right">175.53</td>
<td align="right">0.00</td>
<td align="right">96.00</td>
<td align="right">183.00</td>
<td align="right">309.00</td>
<td align="right">1659.00</td>
<td align="left">▇▂▁▁▁</td>
</tr>
<tr class="even">
<td align="left">nFeature_RNA</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">117.80</td>
<td align="right">70.65</td>
<td align="right">0.00</td>
<td align="right">64.00</td>
<td align="right">107.00</td>
<td align="right">160.00</td>
<td align="right">487.00</td>
<td align="left">▇▇▂▁▁</td>
</tr>
<tr class="odd">
<td align="left">fov</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2.55</td>
<td align="right">1.10</td>
<td align="right">1.00</td>
<td align="right">2.00</td>
<td align="right">3.00</td>
<td align="right">4.00</td>
<td align="right">4.00</td>
<td align="left">▆▇▁▇▇</td>
</tr>
<tr class="even">
<td align="left">cell_ID</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1709.70</td>
<td align="right">1142.42</td>
<td align="right">1.00</td>
<td align="right">751.00</td>
<td align="right">1574.00</td>
<td align="right">2506.00</td>
<td align="right">4984.00</td>
<td align="left">▇▇▆▃▁</td>
</tr>
<tr class="odd">
<td align="left">Area</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3285.18</td>
<td align="right">1549.26</td>
<td align="right">52.00</td>
<td align="right">2170.00</td>
<td align="right">3039.00</td>
<td align="right">4117.00</td>
<td align="right">19223.00</td>
<td align="left">▇▃▁▁▁</td>
</tr>
<tr class="even">
<td align="left">AspectRatio</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1.06</td>
<td align="right">0.33</td>
<td align="right">0.24</td>
<td align="right">0.84</td>
<td align="right">1.01</td>
<td align="right">1.22</td>
<td align="right">4.43</td>
<td align="left">▇▅▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">CenterX_local_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2703.68</td>
<td align="right">1531.36</td>
<td align="right">12.00</td>
<td align="right">1398.00</td>
<td align="right">2702.00</td>
<td align="right">4007.00</td>
<td align="right">5459.00</td>
<td align="left">▇▇▇▇▇</td>
</tr>
<tr class="even">
<td align="left">CenterY_local_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1880.76</td>
<td align="right">948.50</td>
<td align="right">13.00</td>
<td align="right">1146.00</td>
<td align="right">1908.00</td>
<td align="right">2656.00</td>
<td align="right">3633.00</td>
<td align="left">▅▇▇▇▆</td>
</tr>
<tr class="odd">
<td align="left">CenterX_global_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-18292.56</td>
<td align="right">24279.38</td>
<td align="right">-45789.56</td>
<td align="right">-39595.67</td>
<td align="right">-29793.22</td>
<td align="right">8110.11</td>
<td align="right">24783.89</td>
<td align="left">▇▂▁▂▃</td>
</tr>
<tr class="even">
<td align="left">CenterY_global_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">86633.94</td>
<td align="right">73788.99</td>
<td align="right">-1193.67</td>
<td align="right">15457.53</td>
<td align="right">26440.22</td>
<td align="right">160112.89</td>
<td align="right">173213.56</td>
<td align="left">▇▁▁▁▇</td>
</tr>
<tr class="odd">
<td align="left">Width</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">69.79</td>
<td align="right">19.55</td>
<td align="right">17.00</td>
<td align="right">56.00</td>
<td align="right">68.00</td>
<td align="right">81.00</td>
<td align="right">226.00</td>
<td align="left">▃▇▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Height</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">68.64</td>
<td align="right">18.23</td>
<td align="right">13.00</td>
<td align="right">56.00</td>
<td align="right">68.00</td>
<td align="right">80.00</td>
<td align="right">219.00</td>
<td align="left">▂▇▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.MembraneStain</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2548.28</td>
<td align="right">3619.30</td>
<td align="right">0.00</td>
<td align="right">438.00</td>
<td align="right">1085.00</td>
<td align="right">3304.25</td>
<td align="right">38748.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.MembraneStain</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">7224.61</td>
<td align="right">8512.48</td>
<td align="right">0.00</td>
<td align="right">1987.00</td>
<td align="right">3756.00</td>
<td align="right">9045.00</td>
<td align="right">64719.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.PanCK</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3328.65</td>
<td align="right">5991.04</td>
<td align="right">0.00</td>
<td align="right">86.00</td>
<td align="right">341.00</td>
<td align="right">4096.00</td>
<td align="right">58521.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.PanCK</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">9358.86</td>
<td align="right">11533.28</td>
<td align="right">0.00</td>
<td align="right">664.00</td>
<td align="right">2336.00</td>
<td align="right">19091.25</td>
<td align="right">65271.00</td>
<td align="left">▇▃▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.CD45</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">284.49</td>
<td align="right">321.12</td>
<td align="right">0.00</td>
<td align="right">77.00</td>
<td align="right">188.00</td>
<td align="right">400.00</td>
<td align="right">15772.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.CD45</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3821.39</td>
<td align="right">4381.72</td>
<td align="right">0.00</td>
<td align="right">942.00</td>
<td align="right">2101.00</td>
<td align="right">5264.25</td>
<td align="right">57199.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.CD3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">554.76</td>
<td align="right">1792.35</td>
<td align="right">0.00</td>
<td align="right">109.00</td>
<td align="right">239.00</td>
<td align="right">503.00</td>
<td align="right">48588.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.CD3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">5063.40</td>
<td align="right">13134.96</td>
<td align="right">129.00</td>
<td align="right">786.00</td>
<td align="right">1301.00</td>
<td align="right">2129.00</td>
<td align="right">64719.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.DAPI</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">42.30</td>
<td align="right">25.69</td>
<td align="right">0.00</td>
<td align="right">24.00</td>
<td align="right">37.00</td>
<td align="right">55.00</td>
<td align="right">258.00</td>
<td align="left">▇▃▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.DAPI</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">209.63</td>
<td align="right">107.76</td>
<td align="right">0.00</td>
<td align="right">139.00</td>
<td align="right">188.00</td>
<td align="right">255.00</td>
<td align="right">6617.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">nCount_negprobes</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0.99</td>
<td align="right">1.31</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">2.00</td>
<td align="right">15.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">nFeature_negprobes</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0.86</td>
<td align="right">1.07</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">1.00</td>
<td align="right">11.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
</tbody>
</table></div>
<p>Explain: number of rows (68624) represents a cell</p>
<p>Identify important metadata columns and discuss what they mean</p>
<p>This dataset is a subset of the experimental data - only the first 4 fovs of
each, and only the CD and HC sample groups.</p>
<p>seurat fov = slide
bruker cosmx fov = region on slide</p>
<p>View the <code>tissue_sample</code> metadata to prepare for the next step. This represents</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">tissue_sample</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
##  CD_a  CD_b  CD_c  HC_a  HC_b  HC_c 
##  7539 13199 13606  8795 15766  9719</code></pre>
</div>
</div>
<div id="visualising-our-spatial-data" class="section level2" number="6.8">
<h2>
<span class="header-section-number">6.8</span> Visualising our spatial data<a class="anchor" aria-label="anchor" href="#visualising-our-spatial-data"><i class="fas fa-link"></i></a>
</h2>
<p>Visualise clusters in a spatial context</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
## GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b 
##            8795           15766            9719            7539           13199 
## GSM7473690.CD.c 
##           13606</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hc_idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="va">cd_idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="fu">ImageDimPlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="va">hc_idents</span>, group.by <span class="op">=</span> <span class="st">"fov_name"</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-16-1.png" width="672"></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hc_idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="fu">ImageDimPlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="va">hc_idents</span>, group.by <span class="op">=</span> <span class="st">"fov_name"</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-17-1.png" width="672"></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cd_idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span><span class="fu">ImageDimPlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="va">cd_idents</span>, group.by <span class="op">=</span> <span class="st">"fov_name"</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-18-1.png" width="672"></div>
<p>By an individual gene.</p>
<p>Do not need fov_name this time</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ImageFeaturePlot</span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="st">"GSM7473682.HC.a"</span>, features <span class="op">=</span> <span class="st">"S100A6"</span><span class="op">)</span></span></code></pre></div>
<pre class="console"><code>Warning: No layers found matching search pattern provided
Warning in FetchData.Assay5(object = object[[DefaultAssay(object = object)]],  :
  data layer is not found and counts layer is used</code></pre>
<p>Cannot use <code>ImageFeaturePlot</code> yet, without normalisation, dim red etc.</p>
</div>
<div id="count-data-assays" class="section level2" number="6.9">
<h2>
<span class="header-section-number">6.9</span> Count data (<code>assays</code>)<a class="anchor" aria-label="anchor" href="#count-data-assays"><i class="fas fa-link"></i></a>
</h2>
<p>Where are my counts?</p>
<p>TODO: Excalidraw flowchart of assays</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span></span></code></pre></div>
<pre><code>## $RNA
## Assay (v5) data with 980 features for 68624 cells
## First 10 features:
##  TGFB3, PIGR, MALAT1, MZT2A, IL16, XBP1, S100A6, CCL2, CEACAM1, VSIR 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c 
## 
## $negprobes
## Assay (v5) data with 19 features for 68624 cells
## First 10 features:
##  NegPrb19, NegPrb11, NegPrb5, NegPrb16, NegPrb3, NegPrb7, NegPrb12,
## NegPrb8, NegPrb20, NegPrb15 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c</code></pre>
<div id="gene-expression-count-matrices" class="section level3" number="6.9.1">
<h3>
<span class="header-section-number">6.9.1</span> Gene expression count matrices<a class="anchor" aria-label="anchor" href="#gene-expression-count-matrices"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">$</span><span class="va">counts.GSM7473682.HC.a</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgCMatrix"
##        HC_a_1_1 HC_a_2_1 HC_a_3_1 HC_a_4_1 HC_a_5_1
## TGFB3         2        .        .        .        .
## PIGR          3       12        1        2        3
## MALAT1       24       35       10        6       24
## MZT2A        12       11        6       12       17
## IL16          2        .        .        .        .</code></pre>
<p>Row = gene</p>
<p>Column = cell for the sample</p>
</div>
<div id="negative-probes" class="section level3" number="6.9.2">
<h3>
<span class="header-section-number">6.9.2</span> Negative probes<a class="anchor" aria-label="anchor" href="#negative-probes"><i class="fas fa-link"></i></a>
</h3>
<p>Explain that negprobes are negative controls to see how noisy your data is.
These do not bind to anything etc.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span></span></code></pre></div>
<pre><code>## Assay (v5) data with 19 features for 68624 cells
## First 10 features:
##  NegPrb19, NegPrb11, NegPrb5, NegPrb16, NegPrb3, NegPrb7, NegPrb12,
## NegPrb8, NegPrb20, NegPrb15 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">)</span></span></code></pre></div>
<pre><code>##  [1] "NegPrb19" "NegPrb11" "NegPrb5"  "NegPrb16" "NegPrb3"  "NegPrb7" 
##  [7] "NegPrb12" "NegPrb8"  "NegPrb20" "NegPrb15" "NegPrb13" "NegPrb23"
## [13] "NegPrb10" "NegPrb9"  "NegPrb22" "NegPrb14" "NegPrb6"  "NegPrb21"
## [19] "NegPrb18"</code></pre>
<p>19 different negative probes.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code>## Warning in LayerData.Assay5(object = x, layer = i, ...): multiple layers are identified by counts.GSM7473682.HC.a counts.GSM7473683.HC.b counts.GSM7473684.HC.c counts.GSM7473688.CD.a counts.GSM7473689.CD.b counts.GSM7473690.CD.c
##  only the first layer is used</code></pre>
<pre><code>## 19 x 8795 sparse Matrix of class "dgCMatrix"</code></pre>
<pre><code>##   [[ suppressing 33 column names 'HC_a_1_1', 'HC_a_2_1', 'HC_a_3_1' ... ]]</code></pre>
<pre><code>##                                                                           
## NegPrb19 1 1 . . . . . 1 . . . . . . . . . . . . . . . . . . . . . . . 1 .
## NegPrb11 1 . . . . . . . . 2 . . . . . . . . . . . . . . . 1 . 1 . . . . .
## NegPrb5  . 2 . . . 2 . . . . . . . . . . . . . . . . . . 1 . . . . . . . .
## NegPrb16 . 2 . . . 1 . 2 . . . 1 . . . . . . . . . . . . . . . . 1 . . . .
## NegPrb3  . . 1 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb7  . . . . 1 . 1 . . . . . . 1 . . . . . . . . . . . . . . . . . . .
## NegPrb12 . . . . 2 . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb8  . . . . . 2 . . . . 1 . . . . . . . . . . . . . . . . . . . . . 1
## NegPrb20 . . . . . 2 . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb15 . . . . . 1 . . . . . . . 2 . . . . . . . . . . . . . . . . . . .
## NegPrb13 . . . . . . . . 1 . . 1 . . . . . . . . . . . . . . . 1 . . . . .
## NegPrb23 . . . . . . . . . 1 . . . . 1 . . . . . . . . . . . . 1 . . . . 1
## NegPrb10 . . . . . . . . . 1 . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb9  . . . . . . . . . . . . 1 . . . . . . . . . . . . . . . . . . . .
## NegPrb22 . . . . . . . . . . . . . 1 1 . 1 . . . . . . . . . . . . . . . .
## NegPrb14 . . . . . . . . . . . . . . . . . . . . . . . . . 2 . . . . . . .
## NegPrb6  . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb21 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb18 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
##                
## NegPrb19 ......
## NegPrb11 ......
## NegPrb5  ......
## NegPrb16 ......
## NegPrb3  ......
## NegPrb7  ......
## NegPrb12 ......
## NegPrb8  ......
## NegPrb20 ......
## NegPrb15 ......
## NegPrb13 ......
## NegPrb23 ......
## NegPrb10 ......
## NegPrb9  ......
## NegPrb22 ......
## NegPrb14 ......
## NegPrb6  ......
## NegPrb21 ......
## NegPrb18 ......
## 
##  .....suppressing 8762 columns in show(); maybe adjust options(max.print=, width=)
##  ..............................</code></pre>
<p>In the next lesson,</p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="aims.html"><span class="header-section-number">5</span> Aims for today</a></div>
<div class="next"><a href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#understanding-our-spatial-data"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="nav-link" href="#overview-1"><span class="header-section-number">6.1</span> Overview</a></li>
<li><a class="nav-link" href="#how-the-data-was-generated"><span class="header-section-number">6.2</span> How the data was generated</a></li>
<li><a class="nav-link" href="#the-raw-data"><span class="header-section-number">6.3</span> The raw data</a></li>
<li><a class="nav-link" href="#cell-segmentation"><span class="header-section-number">6.4</span> Cell segmentation</a></li>
<li><a class="nav-link" href="#note-on-seuratloadnanostring"><span class="header-section-number">6.5</span> Note on Seurat::LoadNanostring()</a></li>
<li><a class="nav-link" href="#loading-all-samples-from-raw-data"><span class="header-section-number">6.6</span> Loading all samples from raw data</a></li>
<li>
<a class="nav-link" href="#loading-the-workshop-data"><span class="header-section-number">6.7</span> Loading the workshop data</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#understanding-the-seuratobject"><span class="header-section-number">6.7.1</span> Understanding the SeuratObject</a></li></ul>
</li>
<li><a class="nav-link" href="#visualising-our-spatial-data"><span class="header-section-number">6.8</span> Visualising our spatial data</a></li>
<li>
<a class="nav-link" href="#count-data-assays"><span class="header-section-number">6.9</span> Count data (assays)</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#gene-expression-count-matrices"><span class="header-section-number">6.9.1</span> Gene expression count matrices</a></li>
<li><a class="nav-link" href="#negative-probes"><span class="header-section-number">6.9.2</span> Negative probes</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-10.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
