<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 6 Understanding our spatial data | Intro to Spatial Transcriptomics</title>
<meta name="author" content="">
<meta name="description" content="6.1 Learning objectives   6.2 Overview library(Seurat) library(SeuratObject) library(tidyverse) library(here) Conducting processing and data analysis steps with open source, third-party...">
<meta name="generator" content="bookdown 0.42.1 with bs4_book()">
<meta property="og:title" content="Chapter 6 Understanding our spatial data | Intro to Spatial Transcriptomics">
<meta property="og:type" content="book">
<meta property="og:description" content="6.1 Learning objectives   6.2 Overview library(Seurat) library(SeuratObject) library(tidyverse) library(here) Conducting processing and data analysis steps with open source, third-party...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 6 Understanding our spatial data | Intro to Spatial Transcriptomics">
<meta name="twitter:description" content="6.1 Learning objectives   6.2 Overview library(Seurat) library(SeuratObject) library(tidyverse) library(here) Conducting processing and data analysis steps with open source, third-party...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.7.0/transition.js"></script><script src="libs/bs3compat-0.7.0/tabs.js"></script><script src="libs/bs3compat-0.7.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.34.0/datatables.js"></script><link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intro to Spatial Transcriptomics</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Overview</a></li>
<li><a class="" href="intro.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="background.html"><span class="header-section-number">3</span> Background</a></li>
<li><a class="" href="study-overview.html"><span class="header-section-number">4</span> Study overview</a></li>
<li><a class="" href="aims.html"><span class="header-section-number">5</span> Aims for today</a></li>
<li><a class="active" href="understanding-our-spatial-data.html"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="" href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></li>
<li><a class="" href="normalisation-and-scaling.html"><span class="header-section-number">8</span> Normalisation and scaling</a></li>
<li><a class="" href="ReducedDimensionality.html"><span class="header-section-number">9</span> Reduce dimensionality - UMAPs</a></li>
<li><a class="" href="BatchCorrection.html"><span class="header-section-number">10</span> Batch Correction</a></li>
<li><a class="" href="clustering.html"><span class="header-section-number">11</span> Clustering</a></li>
<li><a class="" href="ClusterLabelling.html"><span class="header-section-number">12</span> Clustering Labeling</a></li>
<li><a class="" href="SpatiallyRestrictedGenes.html"><span class="header-section-number">13</span> Spatially Restricted Genes</a></li>
<li><a class="" href="Summary.html"><span class="header-section-number">14</span> Summary</a></li>
<li><a class="" href="NextSteps.html"><span class="header-section-number">15</span> Next steps</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">16</span> Resources</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="understanding-our-spatial-data" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> Understanding our spatial data<a class="anchor" aria-label="anchor" href="#understanding-our-spatial-data"><i class="fas fa-link"></i></a>
</h1>
<div class="inline-figure"><img src="images/workflows/01_counts.png" style="width:100.0%"></div>
<div id="learning-objectives-1" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives-1"><i class="fas fa-link"></i></a>
</h2>
</div>
<div id="overview-1" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> Overview<a class="anchor" aria-label="anchor" href="#overview-1"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.github.io/seurat-object/">SeuratObject</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://here.r-lib.org/">here</a></span><span class="op">)</span></span></code></pre></div>
<p>Conducting processing and data analysis steps with open source, third-party
software/packages provides flexibility and supports a range of different
analyses. However, there are key steps undertaken before you can load your data
into R/Python.</p>
</div>
<div id="the-raw-data" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> The raw data<a class="anchor" aria-label="anchor" href="#the-raw-data"><i class="fas fa-link"></i></a>
</h2>
<div class="inline-figure"><img src="images/30_data/data_generation.png"></div>
<p>The files generated by high-resolution spatial transcriptomics platforms, such
as the CosMx Spatial Molecular Imager (SMI), are typically categorised based on
their processing stage. The <strong>Raw Files</strong> contain the primary output from the
instrument’s run, necessary for decoding and spatial assignment, while the
<strong>Flat Files</strong> represent the aggregated, cell-level data ready for statistical
analysis in R or other environments.</p>
<div id="raw-files" class="section level3" number="6.3.1">
<h3>
<span class="header-section-number">6.3.1</span> Raw files<a class="anchor" aria-label="anchor" href="#raw-files"><i class="fas fa-link"></i></a>
</h3>
<p>The raw files represent the initial, large-volume data output directly from the
imaging instrument after the spatial transcriptomics experiment is complete.
This directory contains all the key imaging and log information required to
computationally process the raw fluorescent signals into <strong>quantifiable transcript
locations and cellular boundaries</strong>.</p>
<p>Key raw folders and files contain essential data used for <strong>segmentation and
cell coordinate mapping</strong>. For example, the <code>CellStatsDir/Morphology2D</code> folder
holds the images, including those generated from immunofluorescence staining
(e.g., DAPI, PanCK, CD45), which are vital inputs for cell segmentation, as the
<strong>raw gene expression measurements are initially at the transcript level, not the
cell level</strong>.</p>
<p>Additionally, the <code>RunSummary/</code> directory contains the instrument configuration
files necessary for accurately converting image pixels into physical slide X-Y
coordinates.</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="35%">
<col width="64%">
</colgroup>
<thead><tr class="header">
<th>Folder / File Name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>Flowcell_Folder/</code></td>
<td>Root directory containing logs and slide/run data</td>
</tr>
<tr class="even">
<td><code>Logs/</code></td>
<td>FOV coordinate files</td>
</tr>
<tr class="odd">
<td><code>GSM7473682_HC_a_S1/</code></td>
<td>Main data directory for slide <code>GSM7473682_HC_a</code>, slot 1</td>
</tr>
<tr class="even">
<td><code>plex_001.csv</code></td>
<td>Target probes metadata</td>
</tr>
<tr class="odd">
<td><code>CellStatsDir/</code></td>
<td>Segmentation outputs (labels, overlays, morphology images)</td>
</tr>
<tr class="even">
<td><code>CellStatsDir/</code></td>
<td>Location of high-resolution images</td>
</tr>
<tr class="odd">
<td><code>RunSummary/</code></td>
<td>Instrument configuration files (e.g. pixel–nm ratio)</td>
</tr>
<tr class="even">
<td><code>AnalysisResults/</code></td>
<td>Processed analysis outputs (FOV summaries, detection limits, coordinates)</td>
</tr>
</tbody>
</table></div>
<p>Data retrieved from
<a href="https://github.com/Nanostring-Biostats/CosMxDACustomModules/blob/main/Export/CosMxDAExportSetup.docx">CosMxDACustomModules</a>.</p>
</div>
<div id="flat-files" class="section level3" number="6.3.2">
<h3>
<span class="header-section-number">6.3.2</span> Flat files<a class="anchor" aria-label="anchor" href="#flat-files"><i class="fas fa-link"></i></a>
</h3>
<div class="rmdimportant">
<p><strong>You do not need to access every one of these files!</strong></p>
<p>Most of what we need to do processing. Typically a directory of flat files is
read in as a whole, and we don’t need to worry too much. But note there are no
images in here.</p>
</div>
<p>The flat files contain the organised, post-processing outputs from a pipeline
(like AtoMx SIP) that are directly ready for loading into R packages such as
<code>Seurat</code>.</p>
<p>These files come after cell segmentation, where the <strong>single-molecule data are
summarised into count matrices at the single-cell level</strong>. The main file,
<code>SLIDE_exprMat_file.csv.gz</code>, contains the counts of genes per cell.</p>
<div class="rmdimportant">
<p><strong>A note on cell segmentation</strong></p>
<p>Cell segmentation is the computational step in imaging-based ST that
identifies the boundaries of individual cells so that RNA molecules can be
correctly assigned to each one.</p>
<p>It uses nuclear (DAPI) and/or membrane stains to group transcript locations
into a cell-by-gene count matrix and cell border shapes (polygons) for
downstream analysis.</p>
<p>Accurate segmentation is essential because it determines how reliably
transcripts are assigned to individual cells, which directly affects
downstream analyses. Commercial platforms like CosMx and Xenium include
segmentation in their processing pipelines.</p>
<p>However, you may choose to re-run segmentation using independent tools such as
Cellpose (image-based) or Baysor (transcript-based or hybrid) to improve
boundary accuracy and adapt to specific tissue types.</p>
</div>
<p>Additionally, key metadata files are read in alongside the gene counts which
inform the spatial coordinates of FoVs (<code>SLIDE_fov_positions_file.csv.gz</code>),
transcripts (<code>SLIDE_tx_file.csv.gz</code>), and segmented cell boundaries
(<code>SLIDE-polygons.csv.gz</code>).</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="44%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th>File name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>SLIDE-polygons.csv.gz</code></td>
<td>Cell borders</td>
</tr>
<tr class="even">
<td><code>SLIDE_exprMat_file.csv.gz</code></td>
<td>Counts of genes per cell (counts matrix)</td>
</tr>
<tr class="odd">
<td><code>SLIDE_fov_positions_file.csv.gz</code></td>
<td>Location of FOVs on slide</td>
</tr>
<tr class="even">
<td><code>SLIDE_metadata_file.csv.gz</code></td>
<td>Cell-level QC metadata</td>
</tr>
<tr class="odd">
<td><code>SLIDE_tx_file.csv.gz</code></td>
<td>Location of individual transcripts</td>
</tr>
</tbody>
</table></div>
<div class="rmdimportant">
<p><strong>Xenium data</strong></p>
<p>For a Xenium slide there is just the one typical output directory; they discuss
the formats in the section on
<a href="https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-archive-data">data archiving</a>,
but again, software tools take the directory as a whole.</p>
</div>
</div>
</div>
<div id="importing-the-flat-files-into-r" class="section level2" number="6.4">
<h2>
<span class="header-section-number">6.4</span> Importing the flat files into R<a class="anchor" aria-label="anchor" href="#importing-the-flat-files-into-r"><i class="fas fa-link"></i></a>
</h2>
<div class="rmdwarning">
<p><strong>A note on <code><a href="https://satijalab.org/seurat/reference/ReadNanostring.html">Seurat::LoadNanostring()</a></code></strong></p>
<p>To load the flat files into R, we can use the <code><a href="https://satijalab.org/seurat/reference/ReadNanostring.html">Seurat::LoadNanostring()</a></code> function.
However, this default method <strong>drops most of the metadata in the Seurat object</strong>.
For example, what field of view (FoV) each cell a member of is missing.</p>
<p>We provide an alternate function. In time, this should be fixed within Seurat.
See comments <a href="https://github.com/satijalab/seurat/discussions/9261">here</a>. This
is an example code snippet using the replacement function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import the custom LoadNanostring() function</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"scripts/LoadNanostring_edited_function.R"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Define the path to the flat files. Note that the input is the whole directory.</span></span>
<span><span class="co"># The function deals with reading and assigning the metadata correctly to the</span></span>
<span><span class="co"># cells and gene counts</span></span>
<span><span class="va">example_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"raw_data"</span>, <span class="st">"GSM7473682_HC_a/"</span><span class="op">)</span></span>
<span><span class="va">example_so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ReadNanostring.html">LoadNanostring</a></span><span class="op">(</span><span class="va">example_path</span>, assay<span class="op">=</span><span class="st">'RNA'</span>, fov<span class="op">=</span><span class="st">"GSM7473682.HC.a"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div id="loading-the-subsampled-workshop-data" class="section level2" number="6.5">
<h2>
<span class="header-section-number">6.5</span> Loading the subsampled workshop data<a class="anchor" aria-label="anchor" href="#loading-the-subsampled-workshop-data"><i class="fas fa-link"></i></a>
</h2>
<p>We use a subsampled data for this workshop, for three healthy control samples (HC)
and three Crohn’s disease samples (CD).</p>
<p>The first four field of views for each sample are used.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Code to subset the data</span></span>
<span><span class="va">seurat_file_00_raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw.RDS"</span><span class="op">)</span></span>
<span><span class="va">so_raw_full</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">seurat_file_00_raw</span><span class="op">)</span></span>
<span><span class="va">so_raw_subset</span> <span class="op">&lt;-</span> <span class="va">so_raw_full</span><span class="op">[</span>, <span class="va">so_raw_full</span><span class="op">$</span><span class="va">group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD"</span>, <span class="st">"HC"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">so_raw_full</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">fov</span> <span class="op">&lt;=</span> <span class="fl">4</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">so_raw_subset</span>, <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the prepared data</span></span>
<span><span class="va">so_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://here.r-lib.org/reference/here.html">here</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE234713_CosMx_IBD_seurat_00_raw_subsampled.RDS"</span><span class="op">)</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">so_path</span><span class="op">)</span></span></code></pre></div>
<div class="rmdimportant">
<p><strong>Analysing large data sets with Sketch-based integration</strong></p>
<p>When analysing your own data, it is often necessary to work with the entire
data set rather than an arbitrary subsample, as simple subsampling can omit
important biological information or spatial regions. However, processing very
large data sets can be computationally demanding.</p>
<div class="inline-figure"><img src="images/30_data/geosketch.png"></div>
<p>Sketch-based analysis offers a practical solution by selecting a representative
subset of cells that preserves the overall structure and diversity of the full
data set. Instead of arbitrarily subsetting by fields of view (FoVs), this
approach uses geometric sketching to capture both common and rare cell
populations. You can <strong>perform initial preprocessing steps on this smaller subset
and then extend the results back to the complete dataset for downstream analyses.</strong></p>
<p>For more details, see Seurat’s sketch-based analysis vignette<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://satijalab.org/seurat/articles/seurat5_sketch_analysis" class="uri"&gt;https://satijalab.org/seurat/articles/seurat5_sketch_analysis&lt;/a&gt;&lt;/p&gt;'><sup>2</sup></a><span class="math inline">\(^,\)</span><a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://satijalab.org/seurat/articles/parsebio_sketch_integration" class="uri"&gt;https://satijalab.org/seurat/articles/parsebio_sketch_integration&lt;/a&gt;&lt;/p&gt;'><sup>3</sup></a> and the
GeoSketch method description<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;&lt;a href="https://cb.csail.mit.edu/geosketch/" class="uri"&gt;https://cb.csail.mit.edu/geosketch/&lt;/a&gt;&lt;/p&gt;'><sup>4</sup></a> for a Python alternative.</p>
</div>
</div>
<div id="visualising-our-spatial-data" class="section level2" number="6.6">
<h2>
<span class="header-section-number">6.6</span> Visualising our spatial data<a class="anchor" aria-label="anchor" href="#visualising-our-spatial-data"><i class="fas fa-link"></i></a>
</h2>
<p>TODO: add note on plotting H&amp;E image - unavailable with this data.</p>
<ul>
<li>Need to import into stutility, which is not compatible with v5 Seurat</li>
<li><a href="https://satijalab.org/seurat/articles/spatial_vignette" class="uri">https://satijalab.org/seurat/articles/spatial_vignette</a></li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hc_idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="va">cd_idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">orig.ident</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">hc_idents</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="va">i</span>, group.by <span class="op">=</span> <span class="st">"fov_name"</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-9-1.png" width="672"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-9-2.png" width="672"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-9-3.png" width="672"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">cd_idents</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ImageDimPlot.html">ImageDimPlot</a></span><span class="op">(</span><span class="va">so</span>, fov <span class="op">=</span> <span class="va">i</span>, group.by <span class="op">=</span> <span class="st">"fov_name"</span>, axes <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-10-1.png" width="672"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-10-2.png" width="672"><img src="IntroSpatialTranscriptomics_files/figure-html/unnamed-chunk-10-3.png" width="672"></p>
<div class="rmdimportant">
<p><strong>Xenium does not have FOVs</strong></p>
<p>Xenium uses a wide-field fluorescence microscope for automated image
acquisition across a larger, continuous imaging area. This approach allows
Xenium to map gene expression over the entire tissue region selected by the
user, rather than stitching together multiple small, distinct FOV images.</p>
</div>
<div id="understanding-the-seuratobject" class="section level3" number="6.6.1">
<h3>
<span class="header-section-number">6.6.1</span> Understanding the <code>SeuratObject</code><a class="anchor" aria-label="anchor" href="#understanding-the-seuratobject"><i class="fas fa-link"></i></a>
</h3>
<div class="rmdnote">
<p><strong>Why Seurat?</strong></p>
<p>Seurat objects are a data structure to store the count data and additional metadata.
As the data is very high dimensional, it is important
that connected information about cells, samples, and data stay connected
during data analysis.</p>
<p>Other frameworks such as <code>scverse</code>, and data classes like <code>anndata</code> and
<code>SpatialFeatureExperiment</code> represent the data differently, but all meet the
needs above. Differences in compatibility with other tools.</p>
<p>Flexibility across single-cell transcriptomics.</p>
</div>
<p>Display the Seurat object.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span></span></code></pre></div>
<pre><code>## An object of class Seurat 
## 999 features across 68624 samples within 2 assays 
## Active assay: RNA (980 features, 0 variable features)
##  6 layers present: counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c, counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c
##  1 other assay present: negprobes
##  6 spatial fields of view present: GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b GSM7473690.CD.c</code></pre>
<p><strong>999 features across 68624 samples with 2 assays</strong></p>
<ul>
<li>980 genes + 19 negprobes</li>
<li>68.6k number of segmented cells analysed</li>
<li>Two different types of measurements, detailed below</li>
</ul>
<p><strong>Active assay: RNA (980 features, 0 variable features):</strong></p>
<ul>
<li>RNA assay: Primary gene expression measurement data (counts)</li>
<li>980 transcripts measured</li>
<li>No variable features as feature selection has not been conducted yet</li>
</ul>
<p><strong>6 layers present: counts.GSM7473682.HC.a, counts.GSM7473683.HC.b,
counts.GSM7473684.HC.c, counts.GSM7473688.CD.a, counts.GSM7473689.CD.b,
counts.GSM7473690.CD.c</strong></p>
<ul>
<li>Each layer represents a sample from either HC or CD donors. Required for
comparison of both groups downstream</li>
</ul>
<p><strong>1 other assay present: negprobes</strong></p>
<p>This secondary assay specifically stores the counts corresponding to negative
control probes. In spatial transcriptomics, tracking these control probes is
crucial for quality control, specificity assessment, and defining background
signal.</p>
<p><strong>6 spatial fields of view present: GSM7473682.HC.a GSM7473683.HC.b GSM7473684.HC.c GSM7473688.CD.a GSM7473689.CD.b GSM7473690.CD.c:</strong></p>
<p>The list of six spatial fields of view (FOVs) explicitly links the spatial/image
information stored in the Seurat object to the six specific samples/layers
identified above. This demonstrates that the data set is ready for spatially
resolved analysis and visualisation, allowing results (like cell clusters or
gene expression) to be mapped back onto the original tissue sections for all
six integrated samples.</p>
<div class="rmdimportant">
<p><strong>CosMx FOVs and sample names</strong></p>
<p>Note that the field of view names are delimited by <code>.</code>. At the time of writing,
underscores will cause some functions to fail silently. For more information,
see <a href="https://github.com/satijalab/seurat/issues/10103">this issue</a></p>
</div>
</div>
<div id="exploring-the-metadata" class="section level3" number="6.6.2">
<h3>
<span class="header-section-number">6.6.2</span> Exploring the metadata<a class="anchor" aria-label="anchor" href="#exploring-the-metadata"><i class="fas fa-link"></i></a>
</h3>
<p>Along with the transcript counts per-cell, metadata tracks other central
information such as its experimental details. This is important so we know which
groups to compare downstream (HC vs. UC). This study has one sample per slide,
but there are typically more. Examples of metadata include:</p>
<ul>
<li>The spatial coordinates of your cells and genes</li>
<li>Segmented cell boundaries</li>
<li>Which tissue sample and FOV data belongs to</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<pre><code>##               orig.ident nCount_RNA nFeature_RNA fov cell_ID  Area AspectRatio
## HC_a_1_1 GSM7473682.HC.a        366          187   1       1  6153        0.67
## HC_a_2_1 GSM7473682.HC.a        805          283   1       2 16178        0.85
## HC_a_3_1 GSM7473682.HC.a        118           73   1       3  3119        1.52
## HC_a_4_1 GSM7473682.HC.a        106           59   1       4  3988        1.52
## HC_a_5_1 GSM7473682.HC.a        462          207   1       5  4773        0.53
## HC_a_6_1 GSM7473682.HC.a       1314          436   1       6 11588        0.87
##          CenterX_local_px CenterY_local_px CenterX_global_px CenterY_global_px
## HC_a_1_1             2119             3443          19274.56          173198.6
## HC_a_2_1             1847             3404          19002.56          173159.6
## HC_a_3_1             1986             3458          19141.56          173213.6
## HC_a_4_1             2362             3454          19517.56          173209.6
## HC_a_5_1             2159             3415          19314.56          173170.6
## HC_a_6_1             1664             3402          18819.56          173157.6
##          Width Height Mean.MembraneStain Max.MembraneStain Mean.PanCK Max.PanCK
## HC_a_1_1    79    118                680              2218      17170     29735
## HC_a_2_1   137    161                615              4453      16775     29702
## HC_a_3_1    79     52                989              2827      24033     29662
## HC_a_4_1    91     60                449              2099      13575     29651
## HC_a_5_1    63    118               1002              2425      24946     29670
## HC_a_6_1   116    134               1024              2320      25700     29712
##          Mean.CD45 Max.CD45 Mean.CD3 Max.CD3 Mean.DAPI Max.DAPI cell
## HC_a_1_1       198     9639      234    1237        20      131  1_1
## HC_a_2_1       222    21371      234    1520        12      122  2_1
## HC_a_3_1       171     6858      298    1721        13       58  3_1
## HC_a_4_1       146      935      206    1402         6       56  4_1
## HC_a_5_1       299     6387      330    1443        22      116  5_1
## HC_a_6_1       354    23243      328    1610        23      113  6_1
##          individual_code tissue_sample group        condition fov_name
## HC_a_1_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_2_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_3_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_4_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_5_1            HC_a          HC_a    HC Healthy controls HC_a_001
## HC_a_6_1            HC_a          HC_a    HC Healthy controls HC_a_001
##          nCount_negprobes nFeature_negprobes
## HC_a_1_1                2                  2
## HC_a_2_1                5                  3
## HC_a_3_1                1                  1
## HC_a_4_1                0                  0
## HC_a_5_1                3                  2
## HC_a_6_1                8                  5</code></pre>
<p>TODO: Define terms like centroids, segments, fov</p>
<ul>
<li>See: <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2#mouse-brain-vizgen-merscope" class="uri">https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2#mouse-brain-vizgen-merscope</a>
</li>
<li>TODO: Excalidraw of these</li>
</ul>
<p>Display the metadata columns:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<pre><code>##  [1] "orig.ident"         "nCount_RNA"         "nFeature_RNA"      
##  [4] "fov"                "cell_ID"            "Area"              
##  [7] "AspectRatio"        "CenterX_local_px"   "CenterY_local_px"  
## [10] "CenterX_global_px"  "CenterY_global_px"  "Width"             
## [13] "Height"             "Mean.MembraneStain" "Max.MembraneStain" 
## [16] "Mean.PanCK"         "Max.PanCK"          "Mean.CD45"         
## [19] "Max.CD45"           "Mean.CD3"           "Max.CD3"           
## [22] "Mean.DAPI"          "Max.DAPI"           "cell"              
## [25] "individual_code"    "tissue_sample"      "group"             
## [28] "condition"          "fov_name"           "nCount_negprobes"  
## [31] "nFeature_negprobes"</code></pre>
<p>There is a lot of information. Use something like skimr for EDA and to
understand each field at a glance</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">skimr</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/skimr/reference/skim.html">skim</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<caption>
<span id="tab:unnamed-chunk-14">Table 6.1: </span>Data summary</caption>
<tbody>
<tr class="odd">
<td align="left">Name</td>
<td align="left"><a href="mailto:so@meta.data" class="email">so@meta.data</a></td>
</tr>
<tr class="even">
<td align="left">Number of rows</td>
<td align="left">68624</td>
</tr>
<tr class="odd">
<td align="left">Number of columns</td>
<td align="left">31</td>
</tr>
<tr class="even">
<td align="left">_______________________</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Column type frequency:</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">character</td>
<td align="left">7</td>
</tr>
<tr class="odd">
<td align="left">numeric</td>
<td align="left">24</td>
</tr>
<tr class="even">
<td align="left">________________________</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Group variables</td>
<td align="left">None</td>
</tr>
</tbody>
</table></div>
<p><strong>Variable type: character</strong></p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="21%">
<col width="13%">
<col width="18%">
<col width="5%">
<col width="5%">
<col width="8%">
<col width="12%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">skim_variable</th>
<th align="right">n_missing</th>
<th align="right">complete_rate</th>
<th align="right">min</th>
<th align="right">max</th>
<th align="right">empty</th>
<th align="right">n_unique</th>
<th align="right">whitespace</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">orig.ident</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">cell</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">18518</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">individual_code</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">tissue_sample</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">group</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">condition</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">15</td>
<td align="right">16</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">fov_name</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">8</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">0</td>
</tr>
</tbody>
</table></div>
<p><strong>Variable type: numeric</strong></p>
<div class="inline-table"><table style="width:100%;" class="table table-sm">
<colgroup>
<col width="16%">
<col width="8%">
<col width="11%">
<col width="8%">
<col width="7%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">skim_variable</th>
<th align="right">n_missing</th>
<th align="right">complete_rate</th>
<th align="right">mean</th>
<th align="right">sd</th>
<th align="right">p0</th>
<th align="right">p25</th>
<th align="right">p50</th>
<th align="right">p75</th>
<th align="right">p100</th>
<th align="left">hist</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">nCount_RNA</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">224.95</td>
<td align="right">175.53</td>
<td align="right">0.00</td>
<td align="right">96.00</td>
<td align="right">183.00</td>
<td align="right">309.00</td>
<td align="right">1659.00</td>
<td align="left">▇▂▁▁▁</td>
</tr>
<tr class="even">
<td align="left">nFeature_RNA</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">117.80</td>
<td align="right">70.65</td>
<td align="right">0.00</td>
<td align="right">64.00</td>
<td align="right">107.00</td>
<td align="right">160.00</td>
<td align="right">487.00</td>
<td align="left">▇▇▂▁▁</td>
</tr>
<tr class="odd">
<td align="left">fov</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2.55</td>
<td align="right">1.10</td>
<td align="right">1.00</td>
<td align="right">2.00</td>
<td align="right">3.00</td>
<td align="right">4.00</td>
<td align="right">4.00</td>
<td align="left">▆▇▁▇▇</td>
</tr>
<tr class="even">
<td align="left">cell_ID</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1709.70</td>
<td align="right">1142.42</td>
<td align="right">1.00</td>
<td align="right">751.00</td>
<td align="right">1574.00</td>
<td align="right">2506.00</td>
<td align="right">4984.00</td>
<td align="left">▇▇▆▃▁</td>
</tr>
<tr class="odd">
<td align="left">Area</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3285.18</td>
<td align="right">1549.26</td>
<td align="right">52.00</td>
<td align="right">2170.00</td>
<td align="right">3039.00</td>
<td align="right">4117.00</td>
<td align="right">19223.00</td>
<td align="left">▇▃▁▁▁</td>
</tr>
<tr class="even">
<td align="left">AspectRatio</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1.06</td>
<td align="right">0.33</td>
<td align="right">0.24</td>
<td align="right">0.84</td>
<td align="right">1.01</td>
<td align="right">1.22</td>
<td align="right">4.43</td>
<td align="left">▇▅▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">CenterX_local_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2703.68</td>
<td align="right">1531.36</td>
<td align="right">12.00</td>
<td align="right">1398.00</td>
<td align="right">2702.00</td>
<td align="right">4007.00</td>
<td align="right">5459.00</td>
<td align="left">▇▇▇▇▇</td>
</tr>
<tr class="even">
<td align="left">CenterY_local_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1880.76</td>
<td align="right">948.50</td>
<td align="right">13.00</td>
<td align="right">1146.00</td>
<td align="right">1908.00</td>
<td align="right">2656.00</td>
<td align="right">3633.00</td>
<td align="left">▅▇▇▇▆</td>
</tr>
<tr class="odd">
<td align="left">CenterX_global_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">-18292.56</td>
<td align="right">24279.38</td>
<td align="right">-45789.56</td>
<td align="right">-39595.67</td>
<td align="right">-29793.22</td>
<td align="right">8110.11</td>
<td align="right">24783.89</td>
<td align="left">▇▂▁▂▃</td>
</tr>
<tr class="even">
<td align="left">CenterY_global_px</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">86633.94</td>
<td align="right">73788.99</td>
<td align="right">-1193.67</td>
<td align="right">15457.53</td>
<td align="right">26440.22</td>
<td align="right">160112.89</td>
<td align="right">173213.56</td>
<td align="left">▇▁▁▁▇</td>
</tr>
<tr class="odd">
<td align="left">Width</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">69.79</td>
<td align="right">19.55</td>
<td align="right">17.00</td>
<td align="right">56.00</td>
<td align="right">68.00</td>
<td align="right">81.00</td>
<td align="right">226.00</td>
<td align="left">▃▇▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Height</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">68.64</td>
<td align="right">18.23</td>
<td align="right">13.00</td>
<td align="right">56.00</td>
<td align="right">68.00</td>
<td align="right">80.00</td>
<td align="right">219.00</td>
<td align="left">▂▇▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.MembraneStain</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2548.28</td>
<td align="right">3619.30</td>
<td align="right">0.00</td>
<td align="right">438.00</td>
<td align="right">1085.00</td>
<td align="right">3304.25</td>
<td align="right">38748.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.MembraneStain</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">7224.61</td>
<td align="right">8512.48</td>
<td align="right">0.00</td>
<td align="right">1987.00</td>
<td align="right">3756.00</td>
<td align="right">9045.00</td>
<td align="right">64719.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.PanCK</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3328.65</td>
<td align="right">5991.04</td>
<td align="right">0.00</td>
<td align="right">86.00</td>
<td align="right">341.00</td>
<td align="right">4096.00</td>
<td align="right">58521.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.PanCK</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">9358.86</td>
<td align="right">11533.28</td>
<td align="right">0.00</td>
<td align="right">664.00</td>
<td align="right">2336.00</td>
<td align="right">19091.25</td>
<td align="right">65271.00</td>
<td align="left">▇▃▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.CD45</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">284.49</td>
<td align="right">321.12</td>
<td align="right">0.00</td>
<td align="right">77.00</td>
<td align="right">188.00</td>
<td align="right">400.00</td>
<td align="right">15772.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.CD45</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3821.39</td>
<td align="right">4381.72</td>
<td align="right">0.00</td>
<td align="right">942.00</td>
<td align="right">2101.00</td>
<td align="right">5264.25</td>
<td align="right">57199.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.CD3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">554.76</td>
<td align="right">1792.35</td>
<td align="right">0.00</td>
<td align="right">109.00</td>
<td align="right">239.00</td>
<td align="right">503.00</td>
<td align="right">48588.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.CD3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">5063.40</td>
<td align="right">13134.96</td>
<td align="right">129.00</td>
<td align="right">786.00</td>
<td align="right">1301.00</td>
<td align="right">2129.00</td>
<td align="right">64719.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">Mean.DAPI</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">42.30</td>
<td align="right">25.69</td>
<td align="right">0.00</td>
<td align="right">24.00</td>
<td align="right">37.00</td>
<td align="right">55.00</td>
<td align="right">258.00</td>
<td align="left">▇▃▁▁▁</td>
</tr>
<tr class="even">
<td align="left">Max.DAPI</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">209.63</td>
<td align="right">107.76</td>
<td align="right">0.00</td>
<td align="right">139.00</td>
<td align="right">188.00</td>
<td align="right">255.00</td>
<td align="right">6617.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="odd">
<td align="left">nCount_negprobes</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0.99</td>
<td align="right">1.31</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">2.00</td>
<td align="right">15.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
<tr class="even">
<td align="left">nFeature_negprobes</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0.86</td>
<td align="right">1.07</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">1.00</td>
<td align="right">11.00</td>
<td align="left">▇▁▁▁▁</td>
</tr>
</tbody>
</table></div>
<p>Explain: number of rows (68624) represents a cell</p>
<p>Identify important metadata columns and discuss what they mean</p>
<p>This dataset is a subset of the experimental data - only the first 4 fovs of
each, and only the CD and HC sample groups.</p>
<p>seurat fov = slide
bruker cosmx fov = region on slide</p>
<p>View the <code>tissue_sample</code> metadata to prepare for the next step. This represents</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">so</span><span class="op">$</span><span class="va">tissue_sample</span><span class="op">)</span></span></code></pre></div>
<pre><code>## 
##  CD_a  CD_b  CD_c  HC_a  HC_b  HC_c 
##  7539 13199 13606  8795 15766  9719</code></pre>
</div>
</div>
<div id="count-data-assays" class="section level2" number="6.7">
<h2>
<span class="header-section-number">6.7</span> Count data (<code>assays</code>)<a class="anchor" aria-label="anchor" href="#count-data-assays"><i class="fas fa-link"></i></a>
</h2>
<div class="inline-figure"><img src="images/30_data/count_assays.png"></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span></span></code></pre></div>
<pre><code>## $RNA
## Assay (v5) data with 980 features for 68624 cells
## First 10 features:
##  TGFB3, PIGR, MALAT1, MZT2A, IL16, XBP1, S100A6, CCL2, CEACAM1, VSIR 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c 
## 
## $negprobes
## Assay (v5) data with 19 features for 68624 cells
## First 10 features:
##  NegPrb19, NegPrb11, NegPrb5, NegPrb16, NegPrb3, NegPrb7, NegPrb12,
## NegPrb8, NegPrb20, NegPrb15 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c</code></pre>
<div id="gene-expression-count-matrices" class="section level3" number="6.7.1">
<h3>
<span class="header-section-number">6.7.1</span> Gene expression count matrices<a class="anchor" aria-label="anchor" href="#gene-expression-count-matrices"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">$</span><span class="va">counts.GSM7473682.HC.a</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class "dgCMatrix"
##        HC_a_1_1 HC_a_2_1 HC_a_3_1 HC_a_4_1 HC_a_5_1
## TGFB3         2        .        .        .        .
## PIGR          3       12        1        2        3
## MALAT1       24       35       10        6       24
## MZT2A        12       11        6       12       17
## IL16          2        .        .        .        .</code></pre>
<p>Row = gene</p>
<p>Column = cell for the sample</p>
</div>
<div id="negative-probes" class="section level3" number="6.7.2">
<h3>
<span class="header-section-number">6.7.2</span> Negative probes<a class="anchor" aria-label="anchor" href="#negative-probes"><i class="fas fa-link"></i></a>
</h3>
<p>Explain that negprobes are negative controls to see how noisy your data is.
These do not bind to anything etc.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span></span></code></pre></div>
<pre><code>## Assay (v5) data with 19 features for 68624 cells
## First 10 features:
##  NegPrb19, NegPrb11, NegPrb5, NegPrb16, NegPrb3, NegPrb7, NegPrb12,
## NegPrb8, NegPrb20, NegPrb15 
## Layers:
##  counts.GSM7473682.HC.a, counts.GSM7473683.HC.b, counts.GSM7473684.HC.c,
## counts.GSM7473688.CD.a, counts.GSM7473689.CD.b, counts.GSM7473690.CD.c</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">)</span></span></code></pre></div>
<pre><code>##  [1] "NegPrb19" "NegPrb11" "NegPrb5"  "NegPrb16" "NegPrb3"  "NegPrb7" 
##  [7] "NegPrb12" "NegPrb8"  "NegPrb20" "NegPrb15" "NegPrb13" "NegPrb23"
## [13] "NegPrb10" "NegPrb9"  "NegPrb22" "NegPrb14" "NegPrb6"  "NegPrb21"
## [19] "NegPrb18"</code></pre>
<p>19 different negative probes.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">negprobes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code>## 19 x 8795 sparse Matrix of class "dgCMatrix"
##                                                                           
## NegPrb19 1 1 . . . . . 1 . . . . . . . . . . . . . . . . . . . . . . . 1 .
## NegPrb11 1 . . . . . . . . 2 . . . . . . . . . . . . . . . 1 . 1 . . . . .
## NegPrb5  . 2 . . . 2 . . . . . . . . . . . . . . . . . . 1 . . . . . . . .
## NegPrb16 . 2 . . . 1 . 2 . . . 1 . . . . . . . . . . . . . . . . 1 . . . .
## NegPrb3  . . 1 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb7  . . . . 1 . 1 . . . . . . 1 . . . . . . . . . . . . . . . . . . .
## NegPrb12 . . . . 2 . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb8  . . . . . 2 . . . . 1 . . . . . . . . . . . . . . . . . . . . . 1
## NegPrb20 . . . . . 2 . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb15 . . . . . 1 . . . . . . . 2 . . . . . . . . . . . . . . . . . . .
## NegPrb13 . . . . . . . . 1 . . 1 . . . . . . . . . . . . . . . 1 . . . . .
## NegPrb23 . . . . . . . . . 1 . . . . 1 . . . . . . . . . . . . 1 . . . . 1
## NegPrb10 . . . . . . . . . 1 . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb9  . . . . . . . . . . . . 1 . . . . . . . . . . . . . . . . . . . .
## NegPrb22 . . . . . . . . . . . . . 1 1 . 1 . . . . . . . . . . . . . . . .
## NegPrb14 . . . . . . . . . . . . . . . . . . . . . . . . . 2 . . . . . . .
## NegPrb6  . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb21 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
## NegPrb18 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
##                
## NegPrb19 ......
## NegPrb11 ......
## NegPrb5  ......
## NegPrb16 ......
## NegPrb3  ......
## NegPrb7  ......
## NegPrb12 ......
## NegPrb8  ......
## NegPrb20 ......
## NegPrb15 ......
## NegPrb13 ......
## NegPrb23 ......
## NegPrb10 ......
## NegPrb9  ......
## NegPrb22 ......
## NegPrb14 ......
## NegPrb6  ......
## NegPrb21 ......
## NegPrb18 ......
## 
##  .....suppressing 8762 columns in show(); maybe adjust options(max.print=, width=)
##  ..............................</code></pre>
</div>
</div>
<div id="loading-in-multiple-samples" class="section level2" number="6.8">
<h2>
<span class="header-section-number">6.8</span> Loading in multiple samples<a class="anchor" aria-label="anchor" href="#loading-in-multiple-samples"><i class="fas fa-link"></i></a>
</h2>
<p>TODO: Example of loading in multiple samples</p>
<p>TODO: Explanation on splitting and merging layers</p>
</div>
<div id="summary" class="section level2" number="6.9">
<h2>
<span class="header-section-number">6.9</span> Summary<a class="anchor" aria-label="anchor" href="#summary"><i class="fas fa-link"></i></a>
</h2>
<p>TODO</p>

</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="aims.html"><span class="header-section-number">5</span> Aims for today</a></div>
<div class="next"><a href="quality-control-and-filtering.html"><span class="header-section-number">7</span> Quality control and filtering</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#understanding-our-spatial-data"><span class="header-section-number">6</span> Understanding our spatial data</a></li>
<li><a class="nav-link" href="#learning-objectives-1"><span class="header-section-number">6.1</span> Learning objectives</a></li>
<li><a class="nav-link" href="#overview-1"><span class="header-section-number">6.2</span> Overview</a></li>
<li>
<a class="nav-link" href="#the-raw-data"><span class="header-section-number">6.3</span> The raw data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#raw-files"><span class="header-section-number">6.3.1</span> Raw files</a></li>
<li><a class="nav-link" href="#flat-files"><span class="header-section-number">6.3.2</span> Flat files</a></li>
</ul>
</li>
<li><a class="nav-link" href="#importing-the-flat-files-into-r"><span class="header-section-number">6.4</span> Importing the flat files into R</a></li>
<li><a class="nav-link" href="#loading-the-subsampled-workshop-data"><span class="header-section-number">6.5</span> Loading the subsampled workshop data</a></li>
<li>
<a class="nav-link" href="#visualising-our-spatial-data"><span class="header-section-number">6.6</span> Visualising our spatial data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#understanding-the-seuratobject"><span class="header-section-number">6.6.1</span> Understanding the SeuratObject</a></li>
<li><a class="nav-link" href="#exploring-the-metadata"><span class="header-section-number">6.6.2</span> Exploring the metadata</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#count-data-assays"><span class="header-section-number">6.7</span> Count data (assays)</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#gene-expression-count-matrices"><span class="header-section-number">6.7.1</span> Gene expression count matrices</a></li>
<li><a class="nav-link" href="#negative-probes"><span class="header-section-number">6.7.2</span> Negative probes</a></li>
</ul>
</li>
<li><a class="nav-link" href="#loading-in-multiple-samples"><span class="header-section-number">6.8</span> Loading in multiple samples</a></li>
<li><a class="nav-link" href="#summary"><span class="header-section-number">6.9</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intro to Spatial Transcriptomics</strong>" was written by . It was last built on 2025-10-19.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
